diff -ru a/COFF/CMakeLists.txt b/COFF/CMakeLists.txt
--- a/lld/COFF/CMakeLists.txt	2022-12-19 20:04:10.737201766 +0000
+++ b/lld/COFF/CMakeLists.txt	2022-12-19 20:04:00.043228253 +0000
@@ -1,6 +1,7 @@
 set(LLVM_TARGET_DEFINITIONS Options.td)
 tablegen(LLVM Options.inc -gen-opt-parser-defs)
 add_public_tablegen_target(COFFOptionsTableGen)
+link_libraries(atomic)
 
 add_lld_library(lldCOFF
   CallGraphSort.cpp
