From ca3d23f274dbd4646596191fea3fddba38e04185 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 5 Dec 2022 19:42:14 +0000
Subject: [PATCH] added atomic libs for arm

---
 SPECS/dhcp.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index c436fc1..434ca97 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -15,7 +15,7 @@
 Summary:  Dynamic host configuration protocol software
 Name:     dhcp
 Version:  4.4.2
-Release:  17.b1%{?dist}
+Release:  17.b1%{?dist}.redsleeve
 
 # NEVER CHANGE THE EPOCH on this package.  The previous maintainer (prior to
 # dcantrell maintaining the package) made incorrect use of the epoch and
@@ -229,6 +229,10 @@ sed -i -e 's|/var/db/|%{_localstatedir}/lib/dhcpd/|g' contrib/dhcp-lease-list.pl
 #libtoolize --copy --force
 autoreconf --verbose --force --install
 
+%ifarch %{arm}
+export LIBS=-latomic
+%endif
+
 CFLAGS="%{optflags} -fno-strict-aliasing -fcommon" \
 %configure \
     --with-srv-lease-file=%{_localstatedir}/lib/dhcpd/dhcpd.leases \
@@ -514,6 +518,9 @@ done
 %endif
 
 %changelog
+* Mon Dec 05 2022 Jacco Ligthart <jacco@redsleeve.org> 4.4.2-17.b1.el9.redsleeve
+- added atomic libs for arm
+
 * Tue Nov 15 2022 CentOS Sources <bugs@centos.org> - 4.4.2-17.b1.el9.centos
 - Apply debranding changes
 
-- 
2.31.1

