From 6eda2a02170f5ca8ae8b429df254e7d90a8ce80f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 13:24:15 +0100
Subject: [PATCH] added atomic libs for arm

---
 SPECS/dhcp.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 9168427..9f49dfd 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -15,7 +15,7 @@
 Summary:              Dynamic host configuration protocol software
 Name:                 dhcp
 Version:              4.4.2
-Release:              19.b1%{?dist}
+Release:              19.b1%{?dist}.redsleeve
 
 # NEVER CHANGE THE EPOCH on this package.  The previous maintainer (prior to
 # dcantrell maintaining the package) made incorrect use of the epoch and
@@ -230,6 +230,10 @@ sed -i -e 's|/var/db/|%{_localstatedir}/lib/dhcpd/|g' contrib/dhcp-lease-list.pl
 #libtoolize --copy --force
 autoreconf --verbose --force --install
 
+%ifarch %{arm}
+export LIBS=-latomic
+%endif
+
 CFLAGS="%{optflags} -fno-strict-aliasing -fcommon" \
 %configure \
     --with-srv-lease-file=%{_localstatedir}/lib/dhcpd/dhcpd.leases \
@@ -509,6 +513,9 @@ done
 %endif
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> 4.4.2-19.b1.el9.redsleeve
+- added atomic libs for arm
+
 * Thu Oct 26 2023 Release Engineering <releng@rockylinux.org> - 4.4.2-19.b1
 - Change bug tracker path
 
-- 
2.39.3

