From 703b881951b512c5d04f1042ec45da9d80c455d0 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 13 May 2026 18:47:19 +0200
Subject: [PATCH] added arm to golang_arches

---
 SPECS/go-rpm-macros.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/go-rpm-macros.spec b/SPECS/go-rpm-macros.spec
index 2e9dd5c..b1b0964 100644
--- a/SPECS/go-rpm-macros.spec
+++ b/SPECS/go-rpm-macros.spec
@@ -11,7 +11,7 @@ Version:   3.6.0
 %global golang_arches   %{ix86} %{golang_arches_future}
 %global gccgo_arches    %{mips}
 %if 0%{?rhel} >= 9
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %endif
 # Go sources can contain arch-specific files and our macros will package the
 # correct files for each architecture. Therefore, move gopath to _libdir and
@@ -37,7 +37,7 @@ Version:   3.6.0
 ExclusiveArch: %{golang_arches} %{gccgo_arches}
 
 Name:      go-rpm-macros
-Release:   14%{?dist}
+Release:   14%{?dist}.redsleeve
 Summary:   Build-stage rpm automation for Go packages
 
 License:   GPLv3+
@@ -263,6 +263,9 @@ sed -i "s,golist ,%{golist_execdir}/golist ,g" \
 %{_rpmluadir}/fedora/srpm/*.lua
 
 %changelog
+* Wed May 13 2026 Jaccco Ligthart <jacco@redsleeve.org> 3.6.0-14.redsleeve
+- added arm to golang_arches
+
 * Fri Apr 17 2026 Alejandro Sáez <asm@redhat.com> - 3.6.0-14
 - Rebuild with latest Go
 - Resolves: RHEL-158726
-- 
2.47.3

