From 7513d65fa152c4cbd6de8788c2f39b7f4a5f5262 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:16:05 +0100
Subject: [PATCH] added arm to golang_arches

---
 SPECS/go-rpm-macros.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/go-rpm-macros.spec b/SPECS/go-rpm-macros.spec
index d99e6bb..ceea60d 100644
--- a/SPECS/go-rpm-macros.spec
+++ b/SPECS/go-rpm-macros.spec
@@ -11,7 +11,7 @@ Version:   3.2.0
 %global golang_arches   %{ix86} %{golang_arches_future}
 %global gccgo_arches    %{mips}
 %if 0%{?rhel} >= 9
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %endif
 # Go sources can contain arch-specific files and our macros will package the
 # correct files for each architecture. Therefore, move gopath to _libdir and
@@ -36,7 +36,7 @@ Version:   3.2.0
 ExclusiveArch: %{golang_arches} %{gccgo_arches}
 
 Name:      go-rpm-macros
-Release:   2%{?dist}
+Release:   2%{?dist}.redsleeve
 Summary:   Build-stage rpm automation for Go packages
 
 License:   GPLv3+
@@ -251,6 +251,9 @@ sed -i "s,golist,%{golist_execdir}/golist,g" %{buildroot}%{_bindir}/go-rpm-integ
 %{_spectemplatedir}/*.spec
 
 %changelog
+* Sat Nov 25 2023 Jaccco Ligthart <jacco@redsleeve.org> 3.2.0-2.redsleeve
+- added arm to golang_arches
+
 * Fri Jul 28 2023 Alejandro Sáez <asm@redhat.com> - 3.2.0-2
 - Add golang_arches_future
 - Resolves: rhbz#2227224
-- 
2.39.3

