From 1a4130a5ef5d4f012e4ba2f7dddf42b78feca9e0 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:56:57 +0100
Subject: [PATCH] do not require bpftool for arm

---
 SPECS/xdp-tools.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/xdp-tools.spec b/SPECS/xdp-tools.spec
index 97fd1cb..57a548c 100644
--- a/SPECS/xdp-tools.spec
+++ b/SPECS/xdp-tools.spec
@@ -1,6 +1,6 @@
 Name:             xdp-tools
 Version:          1.4.0
-Release:          1%{?dist}
+Release:          1%{?dist}.redsleeve
 Summary:          Utilities and example programs for use with XDP
 %global _soversion 1.4.0
 
@@ -22,7 +22,7 @@ BuildRequires:    m4
 BuildRequires:    emacs-nox
 BuildRequires:    wireshark-cli
 
-%ifnarch i686
+%ifnarch i686 %{arm}
 BuildRequires:    bpftool
 %endif
 
@@ -94,7 +94,7 @@ make install V=1
 %{_sbindir}/xdp-filter
 %{_sbindir}/xdp-loader
 %{_sbindir}/xdpdump
-%ifnarch i686
+%ifnarch i686 %{arm}
 %{_sbindir}/xdp-bench
 %{_sbindir}/xdp-monitor
 %{_sbindir}/xdp-trafficgen
@@ -122,6 +122,9 @@ make install V=1
 %{_libdir}/pkgconfig/libxdp.pc
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> 1.4.0-1.redsleeve
+- do not require bpftool for arm
+
 * Thu Jul 6 2023 Toke Høiland-Jørgensen <toke@redhat.com> 1.4.0-1
 - Upstream version bump
 
-- 
2.39.3

