From 1573b886643f6e02de583153b520749677c6c5d5 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 9 Jun 2025 14:22:52 +0200
Subject: [PATCH] do not require bpftool for arm

---
 SPECS/xdp-tools.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/xdp-tools.spec b/SPECS/xdp-tools.spec
index 1e59e5c..bb33d96 100644
--- a/SPECS/xdp-tools.spec
+++ b/SPECS/xdp-tools.spec
@@ -1,6 +1,6 @@
 Name:             xdp-tools
 Version:          1.5.1
-Release:          1%{?dist}
+Release:          1%{?dist}.redsleeve
 Summary:          Utilities and example programs for use with XDP
 %global _soversion 1.5.0
 
@@ -22,7 +22,7 @@ BuildRequires:    m4
 BuildRequires:    emacs-nox
 BuildRequires:    wireshark-cli
 
-%ifnarch i686
+%ifnarch i686 %{arm}
 BuildRequires:    bpftool
 %endif
 
@@ -95,7 +95,7 @@ make install V=1
 %{_sbindir}/xdp-filter
 %{_sbindir}/xdp-loader
 %{_sbindir}/xdpdump
-%ifnarch i686
+%ifnarch i686 %{arm}
 %{_sbindir}/xdp-bench
 %{_sbindir}/xdp-forward
 %{_sbindir}/xdp-monitor
@@ -124,6 +124,9 @@ make install V=1
 %{_libdir}/pkgconfig/libxdp.pc
 
 %changelog
+* Mon Jun 09 2025 Jacco Ligthart <jacco@redsleeve.org> 1.5.1-1.redsleeve
+- do not require bpftool for arm
+
 * Tue Jan 14 2025 Toke Høiland-Jørgensen <toke@redhat.com> 1.5.1-1
 - Upstream version bump
 
-- 
2.43.5

