From 44e5e7e06eaefdfe68915eeeb60ba43d62426ff6 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 31 May 2024 17:25:22 +0200
Subject: [PATCH] do not require bpftool for arm

---
 SPECS/xdp-tools.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/xdp-tools.spec b/SPECS/xdp-tools.spec
index dffd2f4..ca50bd0 100644
--- a/SPECS/xdp-tools.spec
+++ b/SPECS/xdp-tools.spec
@@ -1,6 +1,6 @@
 Name:             xdp-tools
 Version:          1.4.2
-Release:          1%{?dist}
+Release:          1%{?dist}.redsleeve
 Summary:          Utilities and example programs for use with XDP
 %global _soversion 1.4.0
 
@@ -21,7 +21,7 @@ BuildRequires:    m4
 BuildRequires:    emacs-nox
 BuildRequires:    wireshark-cli
 
-%ifnarch i686
+%ifnarch i686 %{arm}
 BuildRequires:    bpftool
 %endif
 
@@ -96,7 +96,7 @@ make install V=1
 %{_sbindir}/xdp-filter
 %{_sbindir}/xdp-loader
 %{_sbindir}/xdpdump
-%ifnarch i686
+%ifnarch i686 %{arm}
 %{_sbindir}/xdp-bench
 %{_sbindir}/xdp-monitor
 %{_sbindir}/xdp-trafficgen
@@ -124,6 +124,9 @@ make install V=1
 %{_libdir}/pkgconfig/libxdp.pc
 
 %changelog
+* Fri May 31 2024 Jacco Ligthart <jacco@redsleeve.org> 1.4.2-1.redsleeve
+- do not require bpftool for arm
+
 * Tue Feb 6 2024 Toke Høiland-Jørgensen <toke@redhat.com> 1.4.2-1
 - Upstream version bump
 
-- 
2.39.3

