From 041625f7fb6428ddbbf2e496dad8a20c9689c8ae Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:10:12 +0100
Subject: [PATCH] patch for all arms in stead of only armv7

---
 SPECS/gjs.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/gjs.spec b/SPECS/gjs.spec
index 3ff27f1..fd6b6d6 100644
--- a/SPECS/gjs.spec
+++ b/SPECS/gjs.spec
@@ -20,7 +20,7 @@
 
 Name:           gjs
 Version:        1.68.6
-Release:        4%{?dist}
+Release:        4%{?dist}.redsleeve
 Summary:        Javascript Bindings for GNOME
 
 # The following files contain code from Mozilla which
@@ -163,7 +163,7 @@ pushd firefox-%{mozjs_version}
 %patch14 -p1
 %patch15 -p1
 
-%ifarch armv7hl
+%ifarch %{arm}
 # Include definitions for user vfp on armv7 as it causes the compilation to fail without them
 # https://bugzilla.mozilla.org/show_bug.cgi?id=1526653
 %patch17 -p1
@@ -302,6 +302,9 @@ popd
 %{_datadir}/installed-tests/
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.68.6-4.redsleeve
+- patch for all arms in stead of only armv7
+
 * Tue May 16 2023 Florian Müllner <fmuellner@redhat.com> - 1.68.6-4
 - Always initialize callback return value
 Resolves: #2196877
-- 
2.39.3

