From d482ceb2a789fb457b32d52d7605e04c9f98f1d4 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 7 Jun 2025 02:34:34 +0200
Subject: [PATCH] patched for armv6

---
 SPECS/rdma-core.spec | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/SPECS/rdma-core.spec b/SPECS/rdma-core.spec
index fd5fd98..7c3d47b 100644
--- a/SPECS/rdma-core.spec
+++ b/SPECS/rdma-core.spec
@@ -1,6 +1,6 @@
 Name: rdma-core
 Version: 54.0
-Release: 1%{?dist}
+Release: 1%{?dist}.redsleeve
 Summary: RDMA core userspace libraries and daemons
 
 # Almost everything is licensed under the OFA dual GPLv2, 2 Clause BSD license
@@ -16,7 +16,7 @@ Patch9999: 9999-udev-keep-NAME_KERNEL-as-default-interface-naming-co.patch
 %define with_static %{?_with_static: 1} %{?!_with_static: 0}
 
 # 32-bit arm is missing required arch-specific memory barriers,
-ExcludeArch: %{arm}
+#ExcludeArch: %{arm}
 
 BuildRequires: binutils
 BuildRequires: cmake >= 2.8.11
@@ -26,7 +26,7 @@ BuildRequires: pkgconfig
 BuildRequires: pkgconfig(libnl-3.0)
 BuildRequires: pkgconfig(libnl-route-3.0)
 BuildRequires: /usr/bin/rst2man
-BuildRequires: valgrind-devel
+#BuildRequires: valgrind-devel
 BuildRequires: systemd
 BuildRequires: systemd-devel
 %if 0%{?fedora} >= 32 || 0%{?rhel} >= 8
@@ -395,7 +395,9 @@ fi
 %config(noreplace) %{_sysconfdir}/rdma/modules/rdma.conf
 %config(noreplace) %{_sysconfdir}/rdma/modules/roce.conf
 %dir %{_sysconfdir}/modprobe.d
+%ifnarch %{arm}
 %config(noreplace) %{_sysconfdir}/modprobe.d/mlx4.conf
+%endif
 %config(noreplace) %{_sysconfdir}/modprobe.d/truescale.conf
 %{_unitdir}/rdma-hw.target
 %{_unitdir}/rdma-load-modules@.service
@@ -432,13 +434,14 @@ fi
 %endif
 %{_libdir}/lib*.so
 %{_libdir}/pkgconfig/*.pc
-%{_mandir}/man3/efadv*
-%{_mandir}/man3/hnsdv*
 %{_mandir}/man3/ibv_*
 %{_mandir}/man3/rdma*
 %{_mandir}/man3/umad*
 %{_mandir}/man3/*_to_ibv_rate.*
 %{_mandir}/man7/rdma_cm.*
+%ifnarch %{arm}
+%{_mandir}/man3/efadv*
+%{_mandir}/man3/hnsdv*
 %{_mandir}/man3/mlx5dv*
 %{_mandir}/man3/mlx4dv*
 %{_mandir}/man3/manadv*
@@ -447,6 +450,7 @@ fi
 %{_mandir}/man7/mlx5dv*
 %{_mandir}/man7/mlx4dv*
 %{_mandir}/man7/manadv*
+%endif
 %{_mandir}/man3/ibnd_*
 
 %files -n infiniband-diags
@@ -520,13 +524,15 @@ fi
 %files -n libibverbs
 %dir %{_sysconfdir}/libibverbs.d
 %dir %{_libdir}/libibverbs
-%{_libdir}/libefa.so.*
-%{_libdir}/libhns.so.*
 %{_libdir}/libibverbs*.so.*
 %{_libdir}/libibverbs/*.so
+%ifnarch %{arm}
+%{_libdir}/libefa.so.*
+%{_libdir}/libhns.so.*
 %{_libdir}/libmana.so.*
 %{_libdir}/libmlx5.so.*
 %{_libdir}/libmlx4.so.*
+%endif
 %config(noreplace) %{_sysconfdir}/libibverbs.d/*.driver
 %doc %{_docdir}/%{name}/libibverbs.md
 
@@ -618,6 +624,9 @@ fi
 %endif
 
 %changelog
+* Fri Jun 06 2024 Jacco Ligthart <jacco@redsleeve.org > - 54.0-1.redsleeve
+- patched for armv6
+
 * Tue Oct 29 2024 Kamal Heib <kheib@redhat.com> - 54.0-1
 - Rebase to upstream release v54.0
 - Resolves: RHEL-52886, RHEL-54647, RHEL-51876
-- 
2.43.5

