From 1e8d43fad829c3e0a6188c2a284a9ffec40573a8 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 22 Dec 2022 22:27:34 +0000
Subject: [PATCH] disabled libompd.so for arm

---
 SPECS/libomp.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libomp.spec b/SPECS/libomp.spec
index 53a2d46..a38ea6c 100644
--- a/SPECS/libomp.spec
+++ b/SPECS/libomp.spec
@@ -11,7 +11,7 @@
 
 Name: libomp
 Version: %{libomp_version}%{?rc_ver:~rc%{rc_ver}}
-Release: 1%{?dist}
+Release: 1%{?dist}.redsleeve
 Summary: OpenMP runtime for clang
 
 License: NCSA
@@ -127,8 +127,8 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %files
 %license LICENSE.TXT
 %{_libdir}/libomp.so
-%{_libdir}/libompd.so
 %ifnarch %{arm}
+%{_libdir}/libompd.so
 %{_libdir}/libarcher.so
 %endif
 %ifnarch %{ix86} %{arm}
@@ -154,6 +154,9 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %{_libexecdir}/tests/libomp/
 
 %changelog
+* Thu Dec 22 2022 Jacco Ligthart <jaccor@redsleeve.org> - 14.0.6-1.redsleeve
+- disabled libompd.so for arm
+
 * Wed Jul 20 2022 Timm Bäder <tbaeder@redhat.com> - 14.0.6-1
 - 14.0.6 Release
 
-- 
2.31.1

