From ca43bc0a79c5208cd7aa536ccdece643e41122db Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:29:06 +0100
Subject: [PATCH] disabled libompd.so for arm

---
 SPECS/libomp.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libomp.spec b/SPECS/libomp.spec
index aaffd7a..21e5d13 100644
--- a/SPECS/libomp.spec
+++ b/SPECS/libomp.spec
@@ -14,7 +14,7 @@
 
 Name: libomp
 Version: %{libomp_version}
-Release: 1%{?dist}
+Release: 1%{?dist}.redsleeve
 Summary: OpenMP runtime for clang
 
 License: NCSA
@@ -139,8 +139,8 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %files
 %license LICENSE.TXT
 %{_libdir}/libomp.so
-%{_libdir}/libompd.so
 %ifnarch %{arm}
+%{_libdir}/libompd.so
 %{_libdir}/libarcher.so
 %endif
 %ifnarch %{ix86} %{arm}
@@ -179,6 +179,9 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %{_libexecdir}/tests/libomp/
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jaccor@redsleeve.org> - 16.0.6-1.redsleeve
+- disabled libompd.so for arm
+
 * Wed Jul 05 2023 Nikita Popov <npopov@redhat.com> - 16.0.6-1
 - Update to LLVM 16.0.6
 
-- 
2.39.3

