From afd923671969ca0b539ee0c949f3af7fbc793e00 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 27 May 2023 00:34:39 +0200
Subject: [PATCH] disabled libompd.so for arm

---
 SPECS/libomp.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libomp.spec b/SPECS/libomp.spec
index e7cfde0..33ea845 100644
--- a/SPECS/libomp.spec
+++ b/SPECS/libomp.spec
@@ -13,7 +13,7 @@
 
 Name: libomp
 Version: %{libomp_version}
-Release: 1%{?dist}
+Release: 1%{?dist}.redsleeve
 Summary: OpenMP runtime for clang
 
 License: NCSA
@@ -130,8 +130,8 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %files
 %license LICENSE.TXT
 %{_libdir}/libomp.so
-%{_libdir}/libompd.so
 %ifnarch %{arm}
+%{_libdir}/libompd.so
 %{_libdir}/libarcher.so
 %endif
 %ifnarch %{ix86} %{arm}
@@ -164,6 +164,9 @@ rm -rf %{buildroot}%{_libdir}/libarcher_static.a
 %{_libexecdir}/tests/libomp/
 
 %changelog
+* Fri May 26 2023 Jacco Ligthart <jaccor@redsleeve.org> - 15.0.7-1.redsleeve
+- disabled libompd.so for arm
+
 * Mon Jan 16 2023 Konrad Kleine <kkleine@redhat.com> - 15.0.7-1
 - Update to LLVM 15.0.7
 
-- 
2.31.1

