From 6466a7154be4a4dbc7fa55422e5d9595d788f15c Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 7 Jun 2025 02:14:21 +0200
Subject: [PATCH] no valgrind on arm

---
 SPECS/libtasn1.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libtasn1.spec b/SPECS/libtasn1.spec
index 07088e6..86f7b2a 100644
--- a/SPECS/libtasn1.spec
+++ b/SPECS/libtasn1.spec
@@ -1,7 +1,7 @@
 Summary:	The ASN.1 library used in GNUTLS
 Name:		libtasn1
 Version:	4.16.0
-Release:	9%{?dist}
+Release:	9%{?dist}.redsleeve
 
 # The libtasn1 library is LGPLv2+, utilities are GPLv3+
 License:	GPLv3+ and LGPLv2+
@@ -18,7 +18,7 @@ BuildRequires:	gnupg2
 BuildRequires:	gcc
 BuildRequires:	bison, pkgconfig, help2man
 BuildRequires:	autoconf, automake, libtool
-BuildRequires:	valgrind-devel
+#BuildRequires:	valgrind-devel
 BuildRequires: make
 # Wildcard bundling exception https://fedorahosted.org/fpc/ticket/174
 Provides: bundled(gnulib) = 20130324
@@ -99,6 +99,9 @@ make check
 
 
 %changelog
+* Fri Jun 06 2025 Jacco Ligthart <jacco@redsleeve.org> - 4.16.0-9.redsleeve
+- no valgrind on arm
+
 * Wed Feb 12 2025 Alexander Sosedkin <asosedki@redhat.com> - 4.16.0-9
 - Backport the fix for CVE-2024-12133 (RHEL-78580)
 
-- 
2.43.5

