From be4fc72611c29a53fa96d182ee49273060f449c2 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 7 Mar 2023 23:52:58 +0000
Subject: [PATCH] added lto to the gcc flags

---
 SPECS/pesign.spec | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/SPECS/pesign.spec b/SPECS/pesign.spec
index 94a6972..7b2ac6d 100644
--- a/SPECS/pesign.spec
+++ b/SPECS/pesign.spec
@@ -3,7 +3,7 @@
 Name:    pesign
 Summary: Signing utility for UEFI binaries
 Version: 115
-Release: 6%{?dist}
+Release: 6%{?dist}.redsleeve
 License: GPL-2.0-only
 URL:     https://github.com/rhboot/pesign
 
@@ -62,6 +62,7 @@ git commit -a -q -m "%{version} baseline."
 git am %{patches} </dev/null
 git config --unset user.email
 git config --unset user.name
+sed -i 's/grecord-gcc-switches/& -flto/' Make.defaults
 
 %build
 # Workaround for mandoc not being packaged
@@ -162,6 +163,9 @@ certutil -d %{_sysconfdir}/pki/pesign/ -X -L > /dev/null
 %{python3_sitelib}/mockbuild/plugins/pesign.*
 
 %changelog
+* Tue Mar 07 2023 Jacco Ligthart <jacco@redsleeve.org> - 115-6.redsleeve
+- added lto to the gcc flags
+
 * Wed Jan 18 2023 Robbie Harwood <rharwood@redhat.com> - 115-6
 - Fix chmod invocation
 - Resolves: CVE-2022-3560
-- 
2.31.1

