From 0e0107644ca554b91676ed7ecd1dd601b598015e Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 16 May 2026 15:15:20 +0200
Subject: [PATCH] add %{arm} to ExclusiveArch

---
 SPECS/java-25-openjdk.spec | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/SPECS/java-25-openjdk.spec b/SPECS/java-25-openjdk.spec
index 3857d01..bdb5336 100644
--- a/SPECS/java-25-openjdk.spec
+++ b/SPECS/java-25-openjdk.spec
@@ -367,6 +367,10 @@
 %global pandoc_available 0
 %endif
 
+%ifarch armv6hl
+%global pandoc_available 0
+%endif
+
 # Priority must be 8 digits in total; up to openjdk 1.8, we were using 18..... so when we moved to 11, we had to add another digit
 %if %is_system_jdk
 # Using 10 digits may overflow the int used for priority, so we combine the patch and build versions
@@ -847,7 +851,7 @@ fi
 %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/*.so
 %ifarch %{share_arches}
 %attr(444, root, root) %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/classes.jsa
-%attr(444, root, root) %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/classes_coh.jsa
+#attr(444, root, root) %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/classes_coh.jsa
 %ifnarch %{ix86} %{arm32}
 %attr(444, root, root) %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/classes_nocoops.jsa
 %attr(444, root, root) %{_jvmdir}/%{sdkdir -- %{?1}}/lib/%{vm_variant}/classes_nocoops_coh.jsa
@@ -1247,11 +1251,11 @@ Provides: java-%{origin}-src%{?1} = %{epoch}:%{version}-%{release}
 %global pkgnameroot java-%{featurever}-%{origin}-portable%{?pkgos:-%{pkgos}}
 
 # Define the architectures on which we build
-ExclusiveArch: %{aarch64} %{ppc64le} s390x x86_64 riscv64
+ExclusiveArch: %{aarch64} %{ppc64le} s390x x86_64 riscv64 %{arm}
 
 Name:    java-%{javaver}-%{origin}
 Version: %{newjavaver}.%{buildver}
-Release: %{?eaprefix}%{rpmrelease}%{?extraver}%{?dist}
+Release: %{?eaprefix}%{rpmrelease}%{?extraver}%{?dist}.redsleeve
 # Equivalent for the portable build
 %global prelease %{?eaprefix}%{portablerelease}%{?extraver}
 # java-1.5.0-ibm from jpackage.org set Epoch to 1 for unknown reasons
@@ -2602,6 +2606,9 @@ exit 0
 %endif
 
 %changelog
+* Sat May 16 2026 Jacco Ligthart <jacco@redsleeve.org> - 25.0.3.0.9-1.redsleeve
+- add %{arm} to ExclusiveArch
+
 * Sat Apr 25 2026 Release Engineering <releng@rockylinux.org> - 1:25.0.3.0.9-1
 - Build for Rocky Linux %{rocky} using our own portable
 
-- 
2.47.3

