--- a/HdrHistogram_c-0.11.0/src/CMakeLists.txt	2020-06-17 07:25:06.000000000 +0000
+++ b/HdrHistogram_c-0.11.0/src/CMakeLists.txt	2022-02-14 08:41:52.424027311 +0000
@@ -39,6 +39,7 @@
             Threads::Threads
             $<$<BOOL:${HAVE_LIBM}>:m>
             $<$<BOOL:${HAVE_LIBRT}>:rt>
+            atomic
             $<$<BOOL:${WIN32}>:ws2_32>)
     target_include_directories(${NAME}
         PUBLIC
