From 28983279f39352ef27841ba41a8c61c82c46dc32 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 19 Jan 2025 13:18:00 +0100
Subject: [PATCH] removed valgrind dependency

---
 SPECS/libstoragemgmt.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/libstoragemgmt.spec b/SPECS/libstoragemgmt.spec
index 5a8ec16..673a57b 100644
--- a/SPECS/libstoragemgmt.spec
+++ b/SPECS/libstoragemgmt.spec
@@ -2,7 +2,7 @@
 
 Name:           libstoragemgmt
 Version:        1.10.1
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        Storage array management library
 License:        LGPL-2.1-or-later
 URL:            https://github.com/libstorage/libstoragemgmt
@@ -33,7 +33,7 @@ BuildRequires:  ledmon-devel
 BuildRequires:  systemd systemd-devel
 
 BuildRequires:  chrpath
-BuildRequires:  valgrind
+#BuildRequires:  valgrind
 
 %description
 The libStorageMgmt library will provide a vendor agnostic open source storage
@@ -168,7 +168,7 @@ plugin selection for locally managed storage.
 %build
 ./autogen.sh
 
-%configure --with-python3 --disable-static
+%configure --with-python3 --disable-static --without-mem-leak-test
 %make_build
 
 %install
@@ -446,6 +446,9 @@ fi
 %{_mandir}/man1/local_lsmplugin.1*
 
 %changelog
+* Sat Jan 04 2025 Jacco Ligthart <jacco@redsleeve.org> - 1.10.1-1.redsleeve
+- removed valgrind dependency
+
 * Mon Jul 15 2024 Tony Asleson <tasleson@redhat.com> - 1.10.1-1
 - Upgrade to 1.10.1
 
-- 
2.43.5

