From 623a0de63a1a95586ae396607f0e8654012ec9ba Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:31:14 +0100
Subject: [PATCH] removed valgrind dependency

---
 SPECS/libstoragemgmt.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/libstoragemgmt.spec b/SPECS/libstoragemgmt.spec
index 7b72cd6..a3f08c4 100644
--- a/SPECS/libstoragemgmt.spec
+++ b/SPECS/libstoragemgmt.spec
@@ -2,7 +2,7 @@
 
 Name:           libstoragemgmt
 Version:        1.9.7
-Release:        2%{?dist}
+Release:        2%{?dist}.redsleeve
 Summary:        Storage array management library
 License:        LGPLv2+
 URL:            https://github.com/libstorage/libstoragemgmt
@@ -32,7 +32,7 @@ BuildRequires:  python3-devel
 BuildRequires:  systemd systemd-devel
 
 BuildRequires:  chrpath
-BuildRequires:  valgrind
+#BuildRequires:  valgrind
 
 %description
 The libStorageMgmt library will provide a vendor agnostic open source storage
@@ -167,7 +167,7 @@ plugin selection for locally managed storage.
 %build
 ./autogen.sh
 
-%configure --with-python3 --disable-static
+%configure --with-python3 --disable-static --without-mem-leak-test
 %make_build
 
 %install
@@ -444,6 +444,9 @@ fi
 %{_mandir}/man1/local_lsmplugin.1*
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.9.7-2.redsleeve
+- removed valgrind dependency
+
 * Tue Apr 18 2023 Tony Asleson <tasleson@redhat.com> - 1.9.7-2
 - FIPS correction ref: https://issues.redhat.com/browse/RHEL-376
 
-- 
2.39.3

