From 933529453b1dce166e456319bdbc6c827fc4af23 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 27 May 2023 00:35:54 +0200
Subject: [PATCH] removed valgrind dependency

---
 SPECS/libstoragemgmt.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/libstoragemgmt.spec b/SPECS/libstoragemgmt.spec
index f9a37d5..e043e8d 100644
--- a/SPECS/libstoragemgmt.spec
+++ b/SPECS/libstoragemgmt.spec
@@ -2,7 +2,7 @@
 
 Name:           libstoragemgmt
 Version:        1.9.5
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        Storage array management library
 License:        LGPLv2+
 URL:            https://github.com/libstorage/libstoragemgmt
@@ -32,7 +32,7 @@ BuildRequires:  python3-devel
 BuildRequires:  systemd systemd-devel
 
 BuildRequires:  chrpath
-BuildRequires:  valgrind
+#BuildRequires:  valgrind
 
 %description
 The libStorageMgmt library will provide a vendor agnostic open source storage
@@ -167,7 +167,7 @@ plugin selection for locally managed storage.
 %build
 ./autogen.sh
 
-%configure --with-python3 --disable-static
+%configure --with-python3 --disable-static --without-mem-leak-test
 %make_build
 
 %install
@@ -444,6 +444,9 @@ fi
 %{_mandir}/man1/local_lsmplugin.1*
 
 %changelog
+* Fri May 26 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.9.5-1.redsleeve
+- removed valgrind dependency
+
 * Thu Oct 27 2022 Tony Asleson <tasleson@redhat.com> - 1.9.5-1
 - Upgrade to 1.9.5
 - Use systemd-sysusers
-- 
2.31.1

