diff --git a/libstdc++-v3/testsuite/26_numerics/gcd/gcd_neg.cc b/libstdc++-v3/testsuite/26_numerics/gcd/gcd_neg.cc
index b0277b3f4d1..425daff317a 100644
--- a/libstdc++-v3/testsuite/26_numerics/gcd/gcd_neg.cc
+++ b/libstdc++-v3/testsuite/26_numerics/gcd/gcd_neg.cc
@@ -53,3 +53,6 @@ test01()
 // { dg-prune-output "does not have integral type" }
 // { dg-prune-output "non-integral type" }
 // { dg-prune-output "invalid specialization" }
+// These show up with -Wp,-D_GLIBCXX_ASSERTIONS.
+// { dg-prune-output "invalid operands" }
+// { dg-prune-output "wrong type argument" }
diff --git a/libstdc++-v3/testsuite/26_numerics/lcm/lcm_neg.cc b/libstdc++-v3/testsuite/26_numerics/lcm/lcm_neg.cc
index 8cabfe2ea2f..a14d373c461 100644
--- a/libstdc++-v3/testsuite/26_numerics/lcm/lcm_neg.cc
+++ b/libstdc++-v3/testsuite/26_numerics/lcm/lcm_neg.cc
@@ -53,3 +53,6 @@ test01()
 // { dg-prune-output "does not have integral type" }
 // { dg-prune-output "non-integral type" }
 // { dg-prune-output "invalid specialization" }
+// These show up with -Wp,-D_GLIBCXX_ASSERTIONS.
+// { dg-prune-output "invalid operands" }
+// { dg-prune-output "wrong type argument" }
