From 968583fa842c9241cbb77c5cf0752a059fbaa2f4 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 4 Feb 2023 01:25:00 +0000
Subject: [PATCH] added flags for armv6

---
 SPECS/kernel.spec | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index a73c186..d7f6256 100755
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -96,7 +96,7 @@ Summary: The Linux kernel
 %global signmodules 1
 
 # Compress modules only for architectures that build modules
-%ifarch noarch
+%ifarch noarch %{arm}
 %global zipmodules 0
 %else
 %global zipmodules 1
@@ -445,7 +445,7 @@ Summary: The Linux kernel
 %define kernel_mflags KALLSYMS_EXTRA_PASS=1
 # we only build headers/perf/tools on the base arm arches
 # just like we used to only build them on i386 for x86
-%ifnarch armv7hl
+%ifnarch armv7hl armv6hl
 %define with_headers 0
 %define with_cross_headers 0
 %endif
@@ -461,6 +461,11 @@ Summary: The Linux kernel
 %define kernel_image arch/arm64/boot/Image.gz
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 # Should make listnewconfig fail if there's config options
 # printed out?
 %if %{nopatches}
@@ -538,7 +543,7 @@ Name: kernel
 License: GPLv2 and Redistributable, no modification permitted
 URL: https://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
 %if 0%{?fedora}
@@ -560,7 +565,7 @@ BuildRequires: kmod, patch, bash, coreutils, tar, git-core, which
 BuildRequires: bzip2, xz, findutils, gzip, m4, perl-interpreter, perl-Carp, perl-devel, perl-generators, make, diffutils, gawk
 BuildRequires: gcc, binutils, redhat-rpm-config, hmaccalc, bison, flex, gcc-c++
 BuildRequires: net-tools, hostname, bc, elfutils-devel
-BuildRequires: dwarves
+#BuildRequires: dwarves
 BuildRequires: python3-devel
 BuildRequires: gcc-plugin-devel
 # glibc-static is required for a consistent build environment (specifically
@@ -2170,7 +2175,7 @@ BuildKernel %make_target %kernel_image %{use_vdso} lpae
 BuildKernel %make_target %kernel_image %{_use_vdso}
 %endif
 
-%ifnarch noarch i686
+%ifnarch noarch i686 %{arm}
 %if !%{with_debug} && !%{with_zfcpdump} && !%{with_pae} && !%{with_up}
 # If only building the user space tools, then initialize the build environment
 # and some variables so that the various userspace tools can be built.
@@ -3017,6 +3022,9 @@ fi
 #
 #
 %changelog
+* Sat Feb 04 2023 Jacco Ligthart <jacco@redsleeve.org> [5.14.0-162.12.1.el9.redsleeve]
+- added flags for armv6
+
 * Tue Dec 20 2022 Patrick Talbert <ptalbert@redhat.com> [5.14.0-162.12.1.el9_1]
 - x86/fpu: Drop fpregs lock before inheriting FPU permissions (Valentin Schneider) [2154407 2153181]
 - hv_netvsc: Fix race between VF offering and VF association message from host (Mohammed Gamal) [2151605 2149277]
-- 
2.31.1

