From 1236c01e87798d2e420c6eb932ece1998e6dc1cb Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Wed, 9 Nov 2022 21:07:37 +0000
Subject: [PATCH] added flags for armv6

---
 SPECS/kernel.spec | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 895cd5a..469a4c9 100755
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -96,7 +96,7 @@ Summary: The Linux kernel
 %global signmodules 1
 
 # Compress modules only for architectures that build modules
-%ifarch noarch
+%ifarch noarch %{arm}
 %global zipmodules 0
 %else
 %global zipmodules 1
@@ -447,7 +447,7 @@ Summary: The Linux kernel
 %define kernel_mflags KALLSYMS_EXTRA_PASS=1
 # we only build headers/perf/tools on the base arm arches
 # just like we used to only build them on i386 for x86
-%ifnarch armv7hl
+%ifnarch armv7hl armv6hl
 %define with_headers 0
 %define with_cross_headers 0
 %endif
@@ -463,6 +463,11 @@ Summary: The Linux kernel
 %define kernel_image arch/arm64/boot/Image.gz
 %endif
 
+%ifarch %{arm}
+%define asmarch arm
+%define hdrarch arm
+%endif
+
 # Should make listnewconfig fail if there's config options
 # printed out?
 %if %{nopatches}
@@ -540,7 +545,7 @@ Name: kernel
 License: GPLv2 and Redistributable, no modification permitted
 URL: https://www.kernel.org/
 Version: %{rpmversion}
-Release: %{pkg_release}
+Release: %{pkg_release}.redsleeve
 # DO NOT CHANGE THE 'ExclusiveArch' LINE TO TEMPORARILY EXCLUDE AN ARCHITECTURE BUILD.
 # SET %%nobuildarches (ABOVE) INSTEAD
 %if 0%{?fedora}
@@ -562,7 +567,7 @@ BuildRequires: kmod, patch, bash, coreutils, tar, git-core, which
 BuildRequires: bzip2, xz, findutils, gzip, m4, perl-interpreter, perl-Carp, perl-devel, perl-generators, make, diffutils, gawk
 BuildRequires: gcc, binutils, redhat-rpm-config, hmaccalc, bison, flex, gcc-c++
 BuildRequires: net-tools, hostname, bc, elfutils-devel
-BuildRequires: dwarves
+#BuildRequires: dwarves
 BuildRequires: python3-devel
 BuildRequires: gcc-plugin-devel
 # glibc-static is required for a consistent build environment (specifically
@@ -2145,7 +2150,7 @@ BuildKernel %make_target %kernel_image %{use_vdso} lpae
 BuildKernel %make_target %kernel_image %{_use_vdso}
 %endif
 
-%ifnarch noarch i686
+%ifnarch noarch i686 %{arm}
 %if !%{with_debug} && !%{with_zfcpdump} && !%{with_pae} && !%{with_up}
 # If only building the user space tools, then initialize the build environment
 # and some variables so that the various userspace tools can be built.
@@ -2960,6 +2965,9 @@ fi
 #
 #
 %changelog
+* Wed Nov 09 2022 Jacco Ligthart <jacco@redsleeve.org> [5.14.0-70.30.1.el9_0.redsleeve]
+- added flags for armv6
+
 * Fri Oct 14 2022 Herton R. Krzesinski <herton@redhat.com> [5.14.0-70.30.1.el9_0]
 - random: trigger reseeding DRBG on more occasions (Daiki Ueno) [2128970 2125257]
 - random: allow reseeding DRBG with getrandom (Daiki Ueno) [2121129 2114854]
-- 
2.31.1

