commit f4f70c2895e3d325188a42c10eb7bb4335be6773
Author: H.J. Lu <hjl.tools@gmail.com>
Date:   Tue Jan 4 06:58:34 2022 -0800

    elf: Add a comment after trailing backslashes

diff --git a/elf/Makefile b/elf/Makefile
index 0cfcf0a61a442c9f..5b9c36bc6f0a3ee5 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -319,6 +319,7 @@ tests-cxx = \
   tst-nodelete \
   tst-unique3 \
   tst-unique4 \
+# tests-cxx
 
 tests += $(if $(CXX),$(tests-cxx))
 tests-internal += loadtest unload unload2 circleload1 \
@@ -583,6 +584,7 @@ modules-names = \
   vismod1 \
   vismod2 \
   vismod3 \
+# modules-names
 
 modules-names-cxx = \
   tst-dlopen-nodelete-reloc-mod1 \
@@ -608,6 +610,7 @@ modules-names-cxx = \
   tst-unique3lib \
   tst-unique3lib2 \
   tst-unique4lib \
+# modules-names-cxx
 
 modules-names += \
   $(if $(CXX),$(modules-names-cxx)) \
@@ -617,6 +620,7 @@ modules-names += \
   $(tst-tls-many-dynamic-modules-dep-bad) \
   $(tlsmod17a-modules) \
   $(tlsmod18a-modules) \
+# modules-names
 
 # Most modules build with _ISOMAC defined, but those filtered out
 # depend on internal headers.
