From cfdd9e7aa45cdc575df237e2d2eee3219a06829b Mon Sep 17 00:00:00 2001
From: Siddhesh Poyarekar <siddhesh@sourceware.org>
Date: Tue, 17 Dec 2024 17:36:36 -0500
Subject: libio: Fix last NULL-as-0 issue in libioP.h

Signed-off-by: Siddhesh Poyarekar <siddhesh@sourceware.org>
Reviewed-by: Maciej W. Rozycki <macro@redhat.com>

diff --git a/libio/libioP.h b/libio/libioP.h
index 70e2bdfc9d..ad45579e13 100644
--- a/libio/libioP.h
+++ b/libio/libioP.h
@@ -933,7 +933,7 @@ extern int _IO_vscanf (const char *, va_list) __THROW;
        { _IO_MAGIC+_IO_LINKED+_IO_IS_FILEBUF+FLAGS, \
 	 NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, NULL, \
 	 NULL, NULL, (FILE *) CHAIN, FD, \
-	 0, _IO_pos_BAD, 0, 0, { 0 }, 0, _IO_pos_BAD, \
+	 0, _IO_pos_BAD, 0, 0, { 0 }, NULL, _IO_pos_BAD, \
 	 NULL, WDP, NULL }
 # endif
 #endif
