commit fd30525eadff6a4b2ac9478bdd6490d0c9c116d9
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Dec 22 23:33:27 2024 +0100

    assert/test-assert-2.c: Include <unistd.h>
    
    For _exit declaration.

diff --git a/assert/test-assert-2.c b/assert/test-assert-2.c
index 99f8f683e87a0c64..3288609ab28701ed 100644
--- a/assert/test-assert-2.c
+++ b/assert/test-assert-2.c
@@ -27,6 +27,7 @@
 #include <dlfcn.h>
 #include <string.h>
 #include <signal.h>
+#include <unistd.h>
 
 #undef NDEBUG
 #include <assert.h>
