diff --git a/relay/dhcrelay.c b/relay/dhcrelay.c
index c4ee5ba..1c6531a 100644
--- a/relay/dhcrelay.c
+++ b/relay/dhcrelay.c
@@ -843,8 +843,8 @@ main(int argc, char **argv) {
 #ifdef HAVE_LIBCAP_NG
 	/* Drop all capabilities */
 	if (!keep_capabilities) {
-		capng_clear(CAPNG_SELECT_BOTH);
-		capng_apply(CAPNG_SELECT_BOTH);
+		capng_clear(CAPNG_SELECT_CAPS);
+		capng_apply(CAPNG_SELECT_CAPS);
 		log_info ("Dropped all capabilities.");
 	}
 #endif
