From 391e21f887abf8c14f7b97f1edcccbced3c5bfcb Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 18 Dec 2022 06:06:10 +0000
Subject: [PATCH] fixed host for armv6hl

---
 SPECS/ocaml.spec | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/SPECS/ocaml.spec b/SPECS/ocaml.spec
index b08c99e..5897e99 100644
--- a/SPECS/ocaml.spec
+++ b/SPECS/ocaml.spec
@@ -31,7 +31,7 @@
 
 Name:           ocaml
 Version:        4.11.1
-Release:        5%{?dist}.2
+Release:        5%{?dist}.2.redsleeve
 
 Summary:        OCaml compiler and programming environment
 
@@ -196,7 +196,12 @@ make=make
     OC_CFLAGS="$CFLAGS" \
     OC_LDFLAGS="$LDFLAGS" \
     --libdir=%{_libdir}/ocaml \
+%ifarch armv6hl
+    AS="as -meabi=5" \
+    --host armv6hl-redhat-linux-gnueabi
+%else
     --host=`./build-aux/config.guess`
+%endif
 $make world
 %if %{native_compiler}
 $make opt
@@ -378,6 +383,9 @@ rm -f $RPM_BUILD_ROOT%{_libdir}/ocaml/eventlog_metadata
 
 
 %changelog
+* Sun Dec 18 2022 Jacco Ligthart <jacco@redssleeve.org> - 4.11.1-5.2.redsleeve
+- fixed host for armv6hl
+
 * Fri Mar 11 2022 Richard W.M. Jones <rjones@redhat.com> - 4.11.1-5.2
 - Rebuild for EPEL
 resolves: rhbz#2060850
-- 
2.31.1

