
diff --git a/ld/testsuite/ld-plugin/lto.exp b/ld/testsuite/ld-plugin/lto.exp
index 3c129760498..dbda6c4465d 100644
--- a/ld/testsuite/ld-plugin/lto.exp
+++ b/ld/testsuite/ld-plugin/lto.exp
@@ -721,7 +721,7 @@ if { [at_least_gcc_version 4 7] } {
 	] \
     ]
     set exec_output [run_host_cmd "sh" \
-				  "-c \"ulimit -n 16; \
+				  "-c \" \
 				   $ar -rc $plug_opt \
 				   tmpdir/libpr23460.a \
 				   tmpdir/pr23460a.o \
-- 
2.38.1

