From 6c0c7d489bdf106d90b300aeb8d042c7b1ad3d2b Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Fri, 24 Nov 2023 08:10:12 +0000
Subject: [PATCH] Fix building for the s390 target with clang

Commit: 6c0c7d489bdf

---
 opcodes/s390-opc.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/opcodes/s390-opc.c b/opcodes/s390-opc.c
index 0427bd8b2e0..cbfdb3df0b7 100644
--- a/opcodes/s390-opc.c
+++ b/opcodes/s390-opc.c
@@ -262,9 +262,10 @@ const struct s390_operand s390_operands[] =
 
 };
 
-static inline void unused_s390_operands_static_asserts(void)
+static inline void ATTRIBUTE_UNUSED
+unused_s390_operands_static_asserts (void)
 {
-  static_assert(ARRAY_SIZE(s390_operands) - 1 == J32_16);
+  static_assert (ARRAY_SIZE (s390_operands) - 1 == J32_16);
 }
 
 /* Macros used to form opcodes.  */
-- 
2.47.0

