--- binutils.orig/bfd/elfnn-aarch64.c	2025-02-07 12:48:33.944978470 +0000
+++ binutils-2.35.2/bfd/elfnn-aarch64.c	2025-02-07 12:48:43.791002138 +0000
@@ -9515,7 +9515,8 @@ elfNN_aarch64_init_small_plt0_entry (bfd
   /* PR 26312: Explicitly set the sh_entsize to 0 so that
      consumers do not think that the section contains fixed
      sized objects.  */
-  elf_section_data (htab->root.splt->output_section)->this_hdr.sh_entsize = 0;
+  if (elf_section_data (htab->root.splt->output_section) != NULL)
+    elf_section_data (htab->root.splt->output_section)->this_hdr.sh_entsize = 0;
 
   plt_got_2nd_ent = (htab->root.sgotplt->output_section->vma
 		  + htab->root.sgotplt->output_offset
