diff -ru binutils-2.30/ld/configure.tgt binutils-2.30-rock/ld/configure.tgt
--- binutils-2.30/ld/configure.tgt	2021-04-25 04:02:21.308705685 +0200
+++ binutils-2.30-rock/ld/configure.tgt	2021-04-25 03:46:48.039031480 +0200
@@ -174,8 +174,8 @@
 			targ_extra_libpath=$targ_extra_emuls
 			;;
 arm*-*-linux-*)		targ_emul=armelf_linux
-			targ_extra_emuls="armelf armelfb armelfb_linux"
-			targ_extra_libpath="armelfb_linux"
+			targ_extra_emuls="armelf armelf_linux_eabi armelfb armelfb_linux"
+			targ_extra_libpath="armelf_linux_eabi armelfb_linux"
 			;;
 arm*b-*-nacl*)		targ_emul=armelfb_nacl
 			targ_extra_emuls="armelf_nacl"
diff -ru binutils-2.30/bfd/doc/Makefile.in binutils-2.30-rock/bfd/doc/Makefile.in
--- binutils-2.30/bfd/doc/Makefile.in	2021-04-25 04:01:40.936807315 +0200
+++ binutils-2.30-rock/bfd/doc/Makefile.in	2021-04-25 03:56:15.013627770 +0200
@@ -171,6 +171,7 @@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+CFLAGS_FOR_BUILD = -march=armv6 -mfpu=vfp -mfloat-abi=hard
 CFLAGS = @CFLAGS@
 COREFILE = @COREFILE@
 COREFLAG = @COREFLAG@
