diff -rup ../binutils.orig/gas/testsuite/gas/s390/esa-g5.d binutils-2.35.2/gas/testsuite/gas/s390/esa-g5.d
--- ../binutils.orig/gas/testsuite/gas/s390/esa-g5.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/esa-g5.d	2024-12-12 10:43:24.821620218 +0000
@@ -1,4 +1,4 @@
-#name: s390 opcode
+#name: s390 opcode (esa g5)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/esa-reloc.d binutils-2.35.2/gas/testsuite/gas/s390/esa-reloc.d
--- ../binutils.orig/gas/testsuite/gas/s390/esa-reloc.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/esa-reloc.d	2024-12-12 10:44:50.261914579 +0000
@@ -1,4 +1,4 @@
-#name: s390 reloc
+#name: s390 reloc (esa reloc)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/esa-z9-109.d binutils-2.35.2/gas/testsuite/gas/s390/esa-z9-109.d
--- ../binutils.orig/gas/testsuite/gas/s390/esa-z9-109.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/esa-z9-109.d	2024-12-12 10:44:35.717864467 +0000
@@ -1,4 +1,4 @@
-#name: s390 opcode
+#name: s390 opcode (esa z9 109)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/esa-z900.d binutils-2.35.2/gas/testsuite/gas/s390/esa-z900.d
--- ../binutils.orig/gas/testsuite/gas/s390/esa-z900.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/esa-z900.d	2024-12-12 11:02:05.545504284 +0000
@@ -1,4 +1,4 @@
-#name: s390 opcode
+#name: s390 opcode (esa z900)
 #objdump: -drw
 
 .*: +file format .*
@@ -7,51 +7,51 @@ Disassembly of section .text:
 
 .* <foo>:
 .*:	c0 f4 00 00 00 00 [	 ]*jg	0 <foo>
-*([\da-f]+):	c0 04 00 00 00 00 [	 ]*jgnop	\1 <foo\+0x\1>
-*([\da-f]+):	c0 14 00 00 00 00 [	 ]*jgo	\1 <foo\+0x\1>
-*([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 34 00 00 00 00 [	 ]*jgnle	\1 <foo\+0x\1>
-*([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 54 00 00 00 00 [	 ]*jgnhe	\1 <foo\+0x\1>
-*([\da-f]+):	c0 64 00 00 00 00 [	 ]*jglh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
-*([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
-*([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
-*([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
-*([\da-f]+):	c0 94 00 00 00 00 [	 ]*jgnlh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 a4 00 00 00 00 [	 ]*jghe	\1 <foo\+0x\1>
-*([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 c4 00 00 00 00 [	 ]*jgle	\1 <foo\+0x\1>
-*([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 e4 00 00 00 00 [	 ]*jgno	\1 <foo\+0x\1>
-*([\da-f]+):	c0 f4 00 00 00 00 [	 ]*jg	\1 <foo\+0x\1>
-*([\da-f]+):	c0 14 00 00 00 00 [	 ]*jgo	\1 <foo\+0x\1>
-*([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 34 00 00 00 00 [	 ]*jgnle	\1 <foo\+0x\1>
-*([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 54 00 00 00 00 [	 ]*jgnhe	\1 <foo\+0x\1>
-*([\da-f]+):	c0 64 00 00 00 00 [	 ]*jglh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
-*([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
-*([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
-*([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
-*([\da-f]+):	c0 94 00 00 00 00 [	 ]*jgnlh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 a4 00 00 00 00 [	 ]*jghe	\1 <foo\+0x\1>
-*([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
-*([\da-f]+):	c0 c4 00 00 00 00 [	 ]*jgle	\1 <foo\+0x\1>
-*([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
-*([\da-f]+):	c0 e4 00 00 00 00 [	 ]*jgno	\1 <foo\+0x\1>
-*([\da-f]+):	c0 f4 00 00 00 00 [	 ]*jg	\1 <foo\+0x\1>
-*([\da-f]+):	c0 65 00 00 00 00 [	 ]*brasl	%r6,\1 <foo\+0x\1>
-*([\da-f]+):	c0 65 00 00 00 00 [	 ]*brasl	%r6,\1 <foo\+0x\1>
+ *([\da-f]+):	c0 04 00 00 00 00 [	 ]*jgnop	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 14 00 00 00 00 [	 ]*jgo	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 34 00 00 00 00 [	 ]*jgnle	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 54 00 00 00 00 [	 ]*jgnhe	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 64 00 00 00 00 [	 ]*jglh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 94 00 00 00 00 [	 ]*jgnlh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 a4 00 00 00 00 [	 ]*jghe	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 c4 00 00 00 00 [	 ]*jgle	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 e4 00 00 00 00 [	 ]*jgno	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 f4 00 00 00 00 [	 ]*jg	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 14 00 00 00 00 [	 ]*jgo	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 24 00 00 00 00 [	 ]*jgh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 34 00 00 00 00 [	 ]*jgnle	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 44 00 00 00 00 [	 ]*jgl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 54 00 00 00 00 [	 ]*jgnhe	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 64 00 00 00 00 [	 ]*jglh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 74 00 00 00 00 [	 ]*jgne	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 84 00 00 00 00 [	 ]*jge	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 94 00 00 00 00 [	 ]*jgnlh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 a4 00 00 00 00 [	 ]*jghe	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 b4 00 00 00 00 [	 ]*jgnl	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 c4 00 00 00 00 [	 ]*jgle	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 d4 00 00 00 00 [	 ]*jgnh	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 e4 00 00 00 00 [	 ]*jgno	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 f4 00 00 00 00 [	 ]*jg	\1 <foo\+0x\1>
+ *([\da-f]+):	c0 65 00 00 00 00 [	 ]*brasl	%r6,\1 <foo\+0x\1>
+ *([\da-f]+):	c0 65 00 00 00 00 [	 ]*brasl	%r6,\1 <foo\+0x\1>
 .*:	01 0b [	 ]*tam
 .*:	01 0c [	 ]*sam24
 .*:	01 0d [	 ]*sam31
@@ -62,7 +62,7 @@ Disassembly of section .text:
 .*:	b9 97 00 69 [	 ]*dlr	%r6,%r9
 .*:	b9 98 00 69 [	 ]*alcr	%r6,%r9
 .*:	b9 99 00 69 [	 ]*slbr	%r6,%r9
-*([\da-f]+):	c0 60 00 00 00 00 [	 ]*larl	%r6,\1 <foo\+0x\1>
+ *([\da-f]+):	c0 60 00 00 00 00 [	 ]*larl	%r6,\1 <foo\+0x\1>
 .*:	e3 65 af ff 00 1e [	 ]*lrv	%r6,4095\(%r5,%r10\)
 .*:	e3 65 af ff 00 1f [	 ]*lrvh	%r6,4095\(%r5,%r10\)
 .*:	e3 65 af ff 00 3e [	 ]*strv	%r6,4095\(%r5,%r10\)
diff -rup ../binutils.orig/gas/testsuite/gas/s390/esa-z990.d binutils-2.35.2/gas/testsuite/gas/s390/esa-z990.d
--- ../binutils.orig/gas/testsuite/gas/s390/esa-z990.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/esa-z990.d	2024-12-12 10:44:05.349759849 +0000
@@ -1,4 +1,4 @@
-#name: s390 opcode
+#name: s390 opcode (esa z990)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-arch12.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch12.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-arch12.d	2024-12-12 10:18:15.923744415 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch12.d	2024-12-12 10:27:23.392254038 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (arch12)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-arch13.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch13.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-arch13.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch13.d	2024-12-12 10:27:23.392254038 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (arch 13)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-arch14.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch14.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-arch14.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch14.d	2024-12-12 10:27:23.392254038 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (arch14)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-arch15.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch15.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-arch15.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-arch15.d	2024-12-12 10:27:23.392254038 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (arch15)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z10.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z10.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z10.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z10.d	2024-12-12 10:27:23.392254038 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z10)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z13.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z13.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z13.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z13.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z13)
 #objdump: -dr
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z196.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z196.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z196.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z196.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z196)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z9-109.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z9-109.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z9-109.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z9-109.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z9 109)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z9-ec.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z9-ec.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z9-ec.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z9-ec.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z9 ec)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z900.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z900.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z900.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z900.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z900)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-z990.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-z990.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-z990.d	2024-12-12 10:18:15.923744415 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-z990.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (z990)
 #objdump: -drw
 
 .*: +file format .*
diff -rup ../binutils.orig/gas/testsuite/gas/s390/zarch-zEC12.d binutils-2.35.2/gas/testsuite/gas/s390/zarch-zEC12.d
--- ../binutils.orig/gas/testsuite/gas/s390/zarch-zEC12.d	2024-12-12 10:18:15.924744418 +0000
+++ binutils-2.35.2/gas/testsuite/gas/s390/zarch-zEC12.d	2024-12-12 10:27:23.393254042 +0000
@@ -1,4 +1,4 @@
-#name: s390x opcode
+#name: s390x opcodes (zEC12)
 #objdump: -dr
 
 .*: +file format .*
