diff -rup binutils.orig/bfd/configure binutils-2.35.2/bfd/configure
--- binutils.orig/bfd/configure	2023-01-19 12:17:21.293513059 +0000
+++ binutils-2.35.2/bfd/configure	2023-01-19 12:27:25.783974084 +0000
@@ -18921,11 +18921,6 @@ _LT_EOF
       esac
     done ;;
     "bfd_stdint.h":C)
-if test "$GCC" = yes; then
-  echo "/* generated for " `$CC --version | sed 1q` "*/" > tmp-stdint.h
-else
-  echo "/* generated for $CC */" > tmp-stdint.h
-fi
 
 sed 's/^ *//' >> tmp-stdint.h <<EOF
 
diff -rup binutils.orig/config/stdint.m4 binutils-2.35.2/config/stdint.m4
--- binutils.orig/config/stdint.m4	2023-01-19 12:17:20.169515897 +0000
+++ binutils-2.35.2/config/stdint.m4	2023-01-19 12:27:02.920032688 +0000
@@ -192,11 +192,6 @@ fi
 
 # ----------------- done all checks, emit header -------------
 AC_CONFIG_COMMANDS(_GCC_STDINT_H, [
-if test "$GCC" = yes; then
-  echo "/* generated for " `$CC --version | sed 1q` "*/" > tmp-stdint.h
-else
-  echo "/* generated for $CC */" > tmp-stdint.h
-fi
 
 sed 's/^ *//' >> tmp-stdint.h <<EOF
 
