--- binutils.orig/binutils/testsuite/lib/binutils-common.exp	2024-06-27 12:43:48.892438898 +0100
+++ binutils-2.35.2/binutils/testsuite/lib/binutils-common.exp	2024-06-27 12:45:12.134877825 +0100
@@ -523,8 +523,13 @@ proc prune_warnings_extra { text } {
 	# The "\\1" is to try to preserve a "\n" but only if necessary.
 	regsub -all "(^|\n)(\[^\n\]*: warning:\[^\n\]*unsupported GNU_PROPERTY_TYPE\[^\n\]*\n?)+" $text "\\1" text
     }
+
     # PR binutils/23898: It is OK to have gaps in build notes.
     regsub -all "(^|\n)(\[^\n\]*: Warning: Gap in build notes detected from\[^\n\]*\n?)+" $text "\\1" text
+
+    # Ignore LTO warnings triggered by configuring with --enable-pgo-build=lto.
+    regsub -all "(^|\n)(\[^\n\]*lto-wrapper: warning: using serial compilation of \[0-9\]+ LTRANS jobs\[^\n\]*\n?)+" $text "\\1" text
+
     return $text
 }
 
