--- binutils.orig/binutils/readelf.c	2021-02-22 10:44:20.142262864 +0000
+++ binutils-2.35.1/binutils/readelf.c	2021-02-22 10:45:25.646842120 +0000
@@ -19285,6 +19285,7 @@ print_gnu_build_attribute_description (E
 
   if (is_open_attr)
     {
+#if 0 /* Suppressed because these gaps are no longer significant.  */
       /* FIXME: Need to properly allow for section alignment.
 	 16 is just the alignment used on x86_64.  */
       if (global_end > 0
@@ -19295,7 +19296,7 @@ print_gnu_build_attribute_description (E
 	  && same_section (filedata, start, global_end))
 	warn (_("Gap in build notes detected from %#lx to %#lx\n"),
 	      global_end + 1, start - 1);
-
+#endif
       printf (_("    Applies to region from %#lx"), start);
       global_offset = start;
 
