From 727964dd52f5e6d1a42cf0a12508bb0ee4e23833 Mon Sep 17 00:00:00 2001
From: Alice Carlotti <alice.carlotti@arm.com>
Date: Sun, 20 Apr 2025 23:20:44 +0100
Subject: [PATCH] aarch64: Add tests for csdb and eret to system.d

---
--- binutils-2.35.2.orig/gas/testsuite/gas/aarch64/system.s	2025-07-22 08:52:03.803039248 +0100
+++ binutils-2.35.2/gas/testsuite/gas/aarch64/system.s	2025-07-22 09:07:13.082237231 +0100
@@ -1,5 +1,6 @@
 	.text
 	drps
+	eret
 
 	//
 	// HINTS
@@ -11,6 +12,7 @@
 	wfi
 	sev
 	sevl
+	csdb
 
 	.macro	all_hints from=0, to=127
 	hint \from
--- binutils-2.35.2.orig/gas/testsuite/gas/aarch64/system.d	2025-07-22 08:52:03.809039283 +0100
+++ binutils-2.35.2/gas/testsuite/gas/aarch64/system.d	2025-07-22 09:07:49.780518903 +0100
@@ -6,12 +6,14 @@ Disassembly of section \.text:
 
 0+ <.*>:
 .*:	d6bf03e0 	drps
+.*:	d69f03e0 	eret
 .*:	d503201f 	nop
 .*:	d503203f 	yield
 .*:	d503205f 	wfe
 .*:	d503207f 	wfi
 .*:	d503209f 	sev
 .*:	d50320bf 	sevl
+.*:	d503229f 	csdb
 .*:	d503201f 	nop
 .*:	d503203f 	yield
 .*:	d503205f 	wfe
