--- binutils.orig/bfd/dwarf2.c	2021-05-18 12:00:10.363028020 +0100
+++ binutils-2.35.2/bfd/dwarf2.c	2021-05-18 12:01:09.930831909 +0100
@@ -1194,7 +1194,7 @@ read_attribute_value (struct attribute *
     case DW_FORM_ref_addr:
       /* DW_FORM_ref_addr is an address in DWARF2, and an offset in
 	 DWARF3.  */
-      if (unit->version == 3 || unit->version == 4)
+      if (unit->version >= 3)
 	{
 	  if (unit->offset_size == 4)
 	    attr->u.val = read_4_bytes (unit->abfd, info_ptr, info_ptr_end);
