From ad1054ada1f474e433c1dc5aa39921c5bdd41a28 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 16 Jul 2022 01:16:04 +0000
Subject: [PATCH] added config options for armv6hl

---
 SPECS/gcc.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/gcc.spec b/SPECS/gcc.spec
index 4fdcde7..d91c2bb 100644
--- a/SPECS/gcc.spec
+++ b/SPECS/gcc.spec
@@ -116,7 +116,7 @@
 Summary: Various compilers (C, C++, Objective-C, ...)
 Name: gcc
 Version: %{gcc_version}
-Release: %{gcc_release}.4%{?dist}
+Release: %{gcc_release}.4%{?dist}.redsleeve
 # libgcc, libgfortran, libgomp, libstdc++ and crtstuff have
 # GCC Runtime Exception.
 License: GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD
@@ -1109,6 +1109,9 @@ CONFIGURE_OPTS="\
 	--with-tune=generic-armv7-a --with-arch=armv7-a \
 	--with-float=hard --with-fpu=vfpv3-d16 --with-abi=aapcs-linux \
 %endif
+%ifarch armv6hl
+	--with-arch=armv6 --with-float=hard --with-fpu=vfp \
+%endif
 %ifarch mips mipsel
 	--with-arch=mips32r2 --with-fp-32=xx \
 %endif
@@ -3271,6 +3274,9 @@ end
 %{ANNOBIN_GCC_PLUGIN_DIR}/gcc-annobin.so.0.0.0
 
 %changelog
+* Fri Jul 15 2022 Jacco Ligthart <jacco@redsleeve.org> 11.2.1-9.4.redsleeve
+- added config options for armv6hl
+
 * Thu Feb 10 2022 Marek Polacek <polacek@redhat.com> 11.2.1-9.4
 - add --enable-host-bind-now, use it (#2044917)
 
-- 
2.27.0

