From efc2e077f363b06ee51569c14584a6d8a5c32e01 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 29 Nov 2025 22:20:37 +0100
Subject: [PATCH] added config options for armv6hl

---
 SPECS/gcc.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/gcc.spec b/SPECS/gcc.spec
index 042758d..b11153a 100644
--- a/SPECS/gcc.spec
+++ b/SPECS/gcc.spec
@@ -129,7 +129,7 @@
 Summary:              Various compilers (C, C++, Objective-C, ...)
 Name:                 gcc
 Version:              %{gcc_version}
-Release:              %{gcc_release}%{?dist}
+Release:              %{gcc_release}%{?dist}.redsleeve
 # libgcc, libgfortran, libgomp, libstdc++ and crtstuff have
 # GCC Runtime Exception.
 License:              GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD
@@ -1279,6 +1279,9 @@ CONFIGURE_OPTS_NATIVE="\
 	--with-tune=generic-armv7-a --with-arch=armv7-a \
 	--with-float=hard --with-fpu=vfpv3-d16 --with-abi=aapcs-linux \
 %endif
+%ifarch armv6hl
+	--with-arch=armv6 --with-float=hard --with-fpu=vfp \
+%endif
 %ifarch mips mipsel
 	--with-arch=mips32r2 --with-fp-32=xx \
 %endif
@@ -3672,6 +3675,9 @@ end
 %endif
 
 %changelog
+* Sat Nov 29 2025 Jacco Ligthart <jacco@redsleeve.org> 11.5.0-11.redsleeve
+- added config options for armv6hl
+
 * Thu Jul 31 2025 Florian Weimer  <fweimer@redhat.com> - 11.5.0-11
 - Adjust glibc32 build dependency (RHEL-105072)
 
-- 
2.47.3

