From 6651a82ab1eae208955c0c6cfe201dd83114fcea Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 26 May 2023 16:11:38 +0200
Subject: [PATCH] added config options for armv6hl

---
 SPECS/gcc.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/gcc.spec b/SPECS/gcc.spec
index 8362d36..f6c40ca 100644
--- a/SPECS/gcc.spec
+++ b/SPECS/gcc.spec
@@ -128,7 +128,7 @@
 Summary: Various compilers (C, C++, Objective-C, ...)
 Name: gcc
 Version: %{gcc_version}
-Release: %{gcc_release}.3%{?dist}
+Release: %{gcc_release}.3%{?dist}.redsleeve
 # libgcc, libgfortran, libgomp, libstdc++ and crtstuff have
 # GCC Runtime Exception.
 License: GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD
@@ -1182,6 +1182,9 @@ CONFIGURE_OPTS_NATIVE="\
 	--with-tune=generic-armv7-a --with-arch=armv7-a \
 	--with-float=hard --with-fpu=vfpv3-d16 --with-abi=aapcs-linux \
 %endif
+%ifarch armv6hl
+	--with-arch=armv6 --with-float=hard --with-fpu=vfp \
+%endif
 %ifarch mips mipsel
 	--with-arch=mips32r2 --with-fp-32=xx \
 %endif
@@ -3572,6 +3575,9 @@ end
 %endif
 
 %changelog
+* Fri May 26 2023 Jacco Ligthart <jacco@redsleeve.org> 11.3.1-4.3.redsleeve
+- added config options for armv6hl
+
 * Wed Dec 21 2022 Marek Polacek <polacek@redhat.com> 11.3.1-4.3
 - compile the cross binaries as PIE/-z now (#2155452)
 
-- 
2.31.1

