From be91eb84c35ee941fb77785eb113d07efff6b36d Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 28 Nov 2022 23:22:42 +0000
Subject: [PATCH] added config options for armv6hl

---
 SPECS/gcc.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/gcc.spec b/SPECS/gcc.spec
index 0fbbd36..21e120c 100644
--- a/SPECS/gcc.spec
+++ b/SPECS/gcc.spec
@@ -116,7 +116,7 @@
 Summary: Various compilers (C, C++, Objective-C, ...)
 Name: gcc
 Version: %{gcc_version}
-Release: %{gcc_release}.1%{?dist}
+Release: %{gcc_release}.1%{?dist}.redsleeve
 # libgcc, libgfortran, libgomp, libstdc++ and crtstuff have
 # GCC Runtime Exception.
 License: GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD
@@ -1112,6 +1112,9 @@ CONFIGURE_OPTS="\
 	--with-tune=generic-armv7-a --with-arch=armv7-a \
 	--with-float=hard --with-fpu=vfpv3-d16 --with-abi=aapcs-linux \
 %endif
+%ifarch armv6hl
+	--with-arch=armv6 --with-float=hard --with-fpu=vfp \
+%endif
 %ifarch mips mipsel
 	--with-arch=mips32r2 --with-fp-32=xx \
 %endif
@@ -3274,6 +3277,9 @@ end
 %{ANNOBIN_GCC_PLUGIN_DIR}/gcc-annobin.so.0.0.0
 
 %changelog
+* Mon Nov 28 2022 Jacco Ligthart <jacco@redsleeve.org> 11.3.1-2.1.redsleeve
+- added config options for armv6hl
+
 * Tue Jul 12 2022 Marek Polacek <polacek@redhat.com> 11.3.1-2.1
 - fix handling of invalid ranges in std::regex (#2106262)
 
-- 
2.31.1

