From 7056fbd6fc8c5599521ccdd3e16db77e00b610bf Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 14 Feb 2025 18:51:52 +0100
Subject: [PATCH] added config options for armv6hl

---
 SPECS/gcc.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/gcc.spec b/SPECS/gcc.spec
index 004003f..9a8579d 100644
--- a/SPECS/gcc.spec
+++ b/SPECS/gcc.spec
@@ -128,7 +128,7 @@
 Summary:              Various compilers (C, C++, Objective-C, ...)
 Name:                 gcc
 Version:              %{gcc_version}
-Release:              %{gcc_release}%{?dist}
+Release:              %{gcc_release}%{?dist}.redsleeve
 # libgcc, libgfortran, libgomp, libstdc++ and crtstuff have
 # GCC Runtime Exception.
 License:              GPLv3+ and GPLv3+ with exceptions and GPLv2+ with exceptions and LGPLv2+ and BSD
@@ -1206,6 +1206,9 @@ CONFIGURE_OPTS_NATIVE="\
 	--with-tune=generic-armv7-a --with-arch=armv7-a \
 	--with-float=hard --with-fpu=vfpv3-d16 --with-abi=aapcs-linux \
 %endif
+%ifarch armv6hl
+	--with-arch=armv6 --with-float=hard --with-fpu=vfp \
+%endif
 %ifarch mips mipsel
 	--with-arch=mips32r2 --with-fp-32=xx \
 %endif
@@ -3599,6 +3602,9 @@ end
 %endif
 
 %changelog
+* Fri Feb 14 2025 Jacco Ligthart <jacco@redsleeve.org> 11.5.0-5.redsleeve
+- added config options for armv6hl
+
 * Fri Feb  7 2025 Marek Polacek <polacek@redhat.com> 11.5.0-5
 - rebuild for CVE-2020-11023 (RHEL-78373)
 
-- 
2.43.5

