From e90ae9076a108b83c645814f75a574c14a5a4b98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alejandro=20S=C3=A1ez?= <asm@redhat.com>
Date: Wed, 27 Aug 2025 16:18:09 +0200
Subject: [PATCH] Revert DWARF5 as default, use DWARF4

---
 src/internal/buildcfg/exp.go | 13 +------------
 1 file changed, 1 insertion(+), 12 deletions(-)

diff --git a/src/internal/buildcfg/exp.go b/src/internal/buildcfg/exp.go
index 689ca8ce58..b2ea97481b 100644
--- a/src/internal/buildcfg/exp.go
+++ b/src/internal/buildcfg/exp.go
@@ -67,24 +67,13 @@ func ParseGOEXPERIMENT(goos, goarch, goexp string) (*ExperimentFlags, error) {
 		regabiSupported = true
 	}
 
-	// Older versions (anything before V16) of dsymutil don't handle
-	// the .debug_rnglists section in DWARF5. See
-	// https://github.com/golang/go/issues/26379#issuecomment-2677068742
-	// for more context. This disables all DWARF5 on mac, which is not
-	// ideal (would be better to disable just for cases where we know
-	// the build will use external linking). In the GOOS=aix case, the
-	// XCOFF format (as far as can be determined) doesn't seem to
-	// support the necessary section subtypes for DWARF-specific
-	// things like .debug_addr (needed for DWARF 5).
-	dwarf5Supported := (goos != "darwin" && goos != "ios" && goos != "aix")
-
 	baseline := goexperiment.Flags{
 		RegabiWrappers:  regabiSupported,
 		RegabiArgs:      regabiSupported,
 		AliasTypeParams: true,
 		SwissMap:        true,
 		SyncHashTrieMap: true,
-		Dwarf5:          dwarf5Supported,
+		Dwarf5:          false,
 	}
 
 	// Start with the statically enabled set of experiments.
-- 
2.51.0

