--- a/clang/lib/Driver/ToolChains/Gnu.cpp	2024-12-30 00:38:04.157102471 +0100
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp	2024-12-30 00:49:46.296960756 +0100
@@ -2461,6 +2461,7 @@
   static const char *const ARMTriples[] = {"arm-linux-gnueabi"};
   static const char *const ARMHFTriples[] = {"arm-linux-gnueabihf",
                                              "armv7hl-redhat-linux-gnueabi",
+                                             "armv6hl-redhat-linux-gnueabihf",
                                              "armv6hl-suse-linux-gnueabi",
                                              "armv7hl-suse-linux-gnueabi"};
   static const char *const ARMebLibDirs[] = {"/lib"};
