diff -up librdkafka-1.6.1/src/rdkafka_conf.c.orig librdkafka-1.6.1/src/rdkafka_conf.c
--- librdkafka-1.6.1/src/rdkafka_conf.c.orig	2022-02-03 12:07:59.066520253 +0100
+++ librdkafka-1.6.1/src/rdkafka_conf.c	        2022-02-03 12:18:06.477192751 +0100
@@ -707,6 +707,7 @@ static const struct rd_kafka_property rd
 	  "security settings for a network connection using TLS or SSL network "
 	  "protocol. See manual page for `ciphers(1)` and "
 	  "`SSL_CTX_set_cipher_list(3).",
+	  .sdef = "PROFILE=SYSTEM",
           _UNSUPPORTED_SSL
 	},
         { _RK_GLOBAL, "ssl.curves.list", _RK_C_STR,
