commit 39b5233271b997811632871e1b6620a89b384fe8
Author: Martin Cermak <mcermak@redhat.com>
Date:   Thu Jan 20 18:18:00 2022 +0100

    Fix python probing rhbz2027683

diff --git a/tapset/python.stp b/tapset/python.stp
new file mode 100644
index 000000000..b5d06fcd9
--- /dev/null
+++ b/tapset/python.stp
@@ -0,0 +1,17 @@
+/* Systemtap tapset to make it easier to trace Python */
+
+/*
+   Define python.function.entry/return:
+*/
+probe python.function.entry = process("/usr/lib*/libpython*.so*").mark("function__entry")
+{
+    filename = user_string($arg1);
+    funcname = user_string($arg2);
+    lineno = $arg3;
+}
+probe python.function.return = process("/usr/lib*/libpython*.so*").mark("function__return")
+{
+    filename = user_string($arg1);
+    funcname = user_string($arg2);
+    lineno = $arg3;
+}
