commit 2c11a115bdbe41a7f064ccd32b41c76c5b7c516d
Author: Martin Cermak <mcermak@redhat.com>
Date:   Wed Dec 1 14:04:57 2021 +0100

    tapset/linux/nfs.stp:  Fix rhbz1972798

diff --git a/tapset/linux/nfs.stp b/tapset/linux/nfs.stp
index 28f0b90ed..232289e3b 100644
--- a/tapset/linux/nfs.stp
+++ b/tapset/linux/nfs.stp
@@ -979,7 +979,7 @@ probe nfs.aop.readpages = kernel.function ("nfs_readpages") !,
         ino = $mapping->host->i_ino
 
         nr_pages = $nr_pages
-        file = $filp
+        file = @choose_defined($filp, $file)
 
         rpages = __nfs_rpages($mapping->host)
         rsize  = __nfs_server_rsize($mapping->host)
