From 5ea8c08de02aeecea20a91a6bc3d0acda61223e8 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 27 May 2023 00:37:35 +0200
Subject: [PATCH] added armv6h

---
 SPECS/openblas.spec | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/SPECS/openblas.spec b/SPECS/openblas.spec
index 681aa93..572f96a 100644
--- a/SPECS/openblas.spec
+++ b/SPECS/openblas.spec
@@ -15,7 +15,7 @@
 
 Name:           openblas
 Version:        0.3.21
-Release:        2%{?dist}
+Release:        2%{?dist}.redsleeve
 Summary:        An optimized BLAS library based on GotoBLAS2
 License:        BSD
 URL:            https://github.com/xianyi/OpenBLAS/
@@ -361,6 +361,9 @@ export AVX="NO_AVX=1 NO_AVX2=1"
 export AVX="NO_AVX2=1"
 %endif
 
+%endif
+%ifarch armv6hl
+TARGET="TARGET=ARMV6 DYNAMIC_ARCH=0"
 %endif
 %ifarch armv7hl
 # ARM v7 still doesn't have runtime cpu detection...
@@ -440,6 +443,9 @@ cp -a %{_includedir}/lapacke %{buildroot}%{_includedir}/%{name}
 # Fix name of libraries: runtime CPU detection has none
 suffix=""
 # but archs that don't have it do have one
+%ifarch armv6hl
+suffix="_armv6"
+%endif
 %ifarch armv7hl
 suffix="_armv7"
 %endif
@@ -649,6 +655,9 @@ rm -rf %{buildroot}%{_libdir}/cmake
 %endif
 
 %changelog
+* Fri May 26 2023 Jacco Ligthart <jacco@redsleeve.org> - 0.3.21-2.redsleeve
+- added armv6hl
+
 * Tue Jan 31 2023 Matej Mužila <mmuzila@redhat.com> - 0.3.21-2
 - Include openblas.pc
   Resolves: #2115737
-- 
2.31.1

