From 84ea6169210f2e5ecb05b443b33eeba489f3ff08 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 21 Jul 2022 13:18:49 +0000
Subject: [PATCH] changed armv7 to arm

---
 SPECS/rust-srpm-macros.spec | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/SPECS/rust-srpm-macros.spec b/SPECS/rust-srpm-macros.spec
index a69e814..ba3e8f5 100644
--- a/SPECS/rust-srpm-macros.spec
+++ b/SPECS/rust-srpm-macros.spec
@@ -1,6 +1,6 @@
 Name:           rust-srpm-macros
 Version:        17
-Release:        4%{?dist}
+Release:        4%{?dist}.redsleeve
 Summary:        RPM macros for building Rust source packages
 
 License:        MIT
@@ -16,6 +16,7 @@ BuildArch:      noarch
 %autosetup -n rust2rpm-v%{version} -p1
 # https://pagure.io/koji/issue/659
 sed -i -e 's/i686/%%{ix86}/' data/macros.rust-srpm
+sed -i -e 's/armv7hl/%%{arm}/' data/macros.rust-srpm
 
 %install
 install -D -p -m 0644 -t %{buildroot}%{_rpmmacrodir} data/macros.rust-srpm
@@ -25,6 +26,9 @@ install -D -p -m 0644 -t %{buildroot}%{_rpmmacrodir} data/macros.rust-srpm
 %{_rpmmacrodir}/macros.rust-srpm
 
 %changelog
+* Thu Jul 21 2022 Jacco Ligthart <jacco@redsleeve.org> - 17-4.redsleeve
+- changed armv7 to arm
+
 * Tue Aug 10 2021 Mohan Boddu <mboddu@redhat.com> - 17-4
 - Rebuilt for IMA sigs, glibc 2.34, aarch64 flags
   Related: rhbz#1991688
-- 
2.31.1

