See https://bugzilla.redhat.com/show_bug.cgi?id=2095749#c3

diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/char/2.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/char/2.cc
index 071597b4dd3..22cfdc7415b 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/char/2.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/char/2.cc
@@ -99,5 +99,7 @@ void test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/char/3.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/char/3.cc
index ec0f943dd73..6c7c0c60ed0 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/char/3.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/char/3.cc
@@ -351,6 +351,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/char/4.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/char/4.cc
index d0619f9de1f..499871bf5b4 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/char/4.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/char/4.cc
@@ -238,6 +238,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/char/71367.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/char/71367.cc
index 9171b7d5a17..9c6d1794991 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/char/71367.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/char/71367.cc
@@ -62,6 +62,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/char/71557.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/char/71557.cc
index 9d004b05e87..9eeff07aaf3 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/char/71557.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/char/71557.cc
@@ -91,6 +91,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/3.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/3.cc
index 45d4edeca00..6e2efad56d4 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/3.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/3.cc
@@ -351,6 +351,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/4.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/4.cc
index 0518842692f..485c1b41d1b 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/4.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/4.cc
@@ -238,6 +238,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71367.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71367.cc
index 456472601be..180255e948e 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71367.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71367.cc
@@ -62,6 +62,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71557.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71557.cc
index 7c5a829fa60..38673aac17b 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71557.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/71557.cc
@@ -91,6 +91,8 @@ test01()
 int
 main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/6.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/6.cc
index e50afa6aa73..2f326f85afe 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/6.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/6.cc
@@ -74,6 +74,8 @@ void test01()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/6.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/6.cc
index f6e32b7f781..54a921e243b 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/6.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/6.cc
@@ -74,6 +74,8 @@ void test01()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_istream/ws/char/lwg415.cc b/libstdc++-v3/testsuite/27_io/basic_istream/ws/char/lwg415.cc
index fe6980dff29..0a033b81138 100644
--- a/libstdc++-v3/testsuite/27_io/basic_istream/ws/char/lwg415.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_istream/ws/char/lwg415.cc
@@ -71,7 +71,9 @@ test03()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
   test03();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_istream/ws/wchar_t/lwg415.cc b/libstdc++-v3/testsuite/27_io/basic_istream/ws/wchar_t/lwg415.cc
index fd040098b40..3adabc60221 100644
--- a/libstdc++-v3/testsuite/27_io/basic_istream/ws/wchar_t/lwg415.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_istream/ws/wchar_t/lwg415.cc
@@ -71,7 +71,9 @@ test03()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
   test03();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ofstream/open/char/noreplace.cc b/libstdc++-v3/testsuite/27_io/basic_ofstream/open/char/noreplace.cc
index e39f5928a1f..22421b158cb 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ofstream/open/char/noreplace.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ofstream/open/char/noreplace.cc
@@ -15,6 +15,7 @@
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
 #if __cpp_lib_ios_noreplace
   std::ios::openmode noreplace = std::ios::noreplace;
 #else
@@ -26,4 +27,5 @@ int main()
   of.close();
   of.open("noreplace", noreplace);
   VERIFY( ! of.is_open() );
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ofstream/open/wchar_t/noreplace.cc b/libstdc++-v3/testsuite/27_io/basic_ofstream/open/wchar_t/noreplace.cc
index 77f11865ac4..07748578417 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ofstream/open/wchar_t/noreplace.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ofstream/open/wchar_t/noreplace.cc
@@ -15,6 +15,7 @@
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
 #if __cpp_lib_ios_noreplace
   std::wios::openmode noreplace = std::wios::noreplace;
 #else
@@ -26,4 +27,5 @@ int main()
   of.close();
   of.open("noreplace", noreplace);
   VERIFY( ! of.is_open() );
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/n3168.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/n3168.cc
index 12da0b1f011..689ee0ee85b 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/n3168.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/n3168.cc
@@ -98,6 +98,8 @@ test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/n3168.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/n3168.cc
index 652f46f615c..abe55ca1285 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/n3168.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/n3168.cc
@@ -96,6 +96,8 @@ test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/n3168.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/n3168.cc
index 8c2fe85db36..92fb45495c1 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/n3168.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/n3168.cc
@@ -59,6 +59,8 @@ test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/n3168.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/n3168.cc
index 887a9a773d1..efb0c2e878a 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/n3168.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/n3168.cc
@@ -59,6 +59,8 @@ test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_time/char/2.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_time/char/2.cc
index 39ff1761c55..9583c630fa8 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_time/char/2.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_time/char/2.cc
@@ -91,6 +91,8 @@ static bool ampm_time_format()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test02();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_time/wchar_t/2.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_time/wchar_t/2.cc
index 228d61f1e41..a1c3265155f 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_time/wchar_t/2.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_time/wchar_t/2.cc
@@ -91,6 +91,8 @@ static bool ampm_time_format()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test02();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/2.cc b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/2.cc
index adbf7e7e733..4d6b4e2ec96 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/2.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get/wchar_t/2.cc
@@ -99,5 +99,7 @@ void test02()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test02();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/flush/char/2.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/flush/char/2.cc
index de7a371c1c4..d5fc73d1a36 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/flush/char/2.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/flush/char/2.cc
@@ -59,7 +59,9 @@ void test02()
 
   VERIFY( os.good() );
   VERIFY( buf.sync_called() );
+#if __LIBSTDCXX_SO_VERSION >= 8
   VERIFY( buf_tie.sync_called() );
+#endif
 }
 
 void
@@ -80,7 +82,9 @@ test03()
 
   VERIFY( os.rdstate() & std::ios_base::eofbit );
   VERIFY( !buf.sync_called() );
+#if __LIBSTDCXX_SO_VERSION >= 8
   VERIFY( !buf_tie.sync_called() );
+#endif
 }
 
 int main()
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/flush/wchar_t/2.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/flush/wchar_t/2.cc
index b330be46afa..40b9dfdb61b 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/flush/wchar_t/2.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/flush/wchar_t/2.cc
@@ -57,7 +57,9 @@ void test02()
 
   VERIFY( os.good() );
   VERIFY( buf.sync_called() );
+#if __LIBSTDCXX_SO_VERSION >= 8
   VERIFY( buf_tie.sync_called() );
+#endif
 }
 
 void
@@ -78,7 +80,9 @@ test03()
 
   VERIFY( os.rdstate() & std::ios_base::eofbit );
   VERIFY( !buf.sync_called() );
+#if __LIBSTDCXX_SO_VERSION >= 8
   VERIFY( !buf_tie.sync_called() );
+#endif
 }
 
 int main()
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/exceptions_badbit_throw.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/exceptions_badbit_throw.cc
index 138648f5264..70f651818ba 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/exceptions_badbit_throw.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/char/exceptions_badbit_throw.cc
@@ -73,7 +73,9 @@ void test02()
 // libstdc++/9546
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/exceptions_badbit_throw.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/exceptions_badbit_throw.cc
index 731df74de9f..b6d31711910 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/exceptions_badbit_throw.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/seekp/wchar_t/exceptions_badbit_throw.cc
@@ -73,7 +73,9 @@ void test02()
 // libstdc++/9546
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
   test02();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/exceptions_badbit_throw.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/exceptions_badbit_throw.cc
index 2eccfdceb94..98d57ff4259 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/exceptions_badbit_throw.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/char/exceptions_badbit_throw.cc
@@ -47,6 +47,8 @@ void test01()
 // libstdc++/9546
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/exceptions_badbit_throw.cc b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/exceptions_badbit_throw.cc
index 92161b22632..9d75c2c69fb 100644
--- a/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/exceptions_badbit_throw.cc
+++ b/libstdc++-v3/testsuite/27_io/basic_ostream/tellp/wchar_t/exceptions_badbit_throw.cc
@@ -47,6 +47,8 @@ void test01()
 // libstdc++/9546
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
   return 0;
 }
diff --git a/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/char/2.cc b/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/char/2.cc
index d4cd0dc3040..a60acf00b04 100644
--- a/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/char/2.cc
+++ b/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/char/2.cc
@@ -44,5 +44,7 @@ void test01()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
 }
diff --git a/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/wchar_t/2.cc b/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/wchar_t/2.cc
index c639ce54906..5c749922016 100644
--- a/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/wchar_t/2.cc
+++ b/libstdc++-v3/testsuite/27_io/manipulators/extended/get_time/wchar_t/2.cc
@@ -45,5 +45,7 @@ void test01()
 
 int main()
 {
+#if __LIBSTDCXX_SO_VERSION >= 8
   test01();
+#endif
 }
