diff -ru a/libstdc++-v3/src/nonshared98/char8_t-rtti.S b/libstdc++-v3/src/nonshared98/char8_t-rtti.S
--- a/libstdc++-v3/src/nonshared98/char8_t-rtti.S	2023-11-27 20:44:36.689971801 +0100
+++ b/libstdc++-v3/src/nonshared98/char8_t-rtti.S	2023-11-27 22:00:36.179962970 +0100
@@ -24,6 +24,11 @@
     || defined __i386__ || defined __powerpc__ || defined __s390__
 #ifdef __i386__
 #define ALIGN1	.align 4
+#elif defined __arm__
+#define ALIGN1 .align 4
+#define OBJECT %object
+#define SECTION1(x).section .gnu.linkonce.d.rel.ro.x,"aw",%progbits
+#define SECTION3(x).section .gnu.linkonce.r.x,"a",%progbits
 #elif defined __x86_64__
 #define ALIGN1	.align 32
 #define ALIGN2	.align 16
@@ -163,4 +168,4 @@
 _ZTSDu:
 	STRING	"Du"
 #endif
-	.section	.note.GNU-stack,"",@progbits
+	.section	.note.GNU-stack,"",%progbits
diff -ru a/libstdc++-v3/src/nonshared98/extfloat.S b/libstdc++-v3/src/nonshared98/extfloat.S
--- a/libstdc++-v3/src/nonshared98/extfloat.S	2023-11-27 20:44:36.689971801 +0100
+++ b/libstdc++-v3/src/nonshared98/extfloat.S	2023-11-27 22:10:23.846629652 +0100
@@ -22,6 +22,11 @@
 
 #ifdef __i386__
 #define ALIGN1	.align 4
+#elif defined __arm__
+#define ALIGN1 .align 4
+#define OBJECT %object
+#define SECTION1(x).section .gnu.linkonce.d.rel.ro.x,"aw",%progbits
+#define SECTION3(x).section .gnu.linkonce.r.x,"a",%progbits
 #elif defined __x86_64__
 #define ALIGN1	.align 32
 #define ALIGN2	.align 16
@@ -544,4 +549,4 @@
 	.size	SYM(_ZTSDF128_), 7
 _ZTSDF128_:
 	STRING	"DF128_"
-	.section	.note.GNU-stack,"",@progbits
+	.section	.note.GNU-stack,"",%progbits
