From 119eb590f5f0b89611d46cdec805b22767f8a6c0 Mon Sep 17 00:00:00 2001
From: Pingfan Liu <piliu@redhat.com>
Date: Wed, 17 Apr 2024 10:36:34 +0800
Subject: [PATCH 3/8] numademo: Fix the using of the uninitialized value

Signed-off-by: Pingfan Liu <piliu@redhat.com>
---
 numademo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/numademo.c b/numademo.c
index 355b269..6f076ee 100644
--- a/numademo.c
+++ b/numademo.c
@@ -250,6 +250,8 @@ static void memtest(char *name, unsigned char *mem)
 
 #endif
 		default:
+			gettimeofday(&start,NULL);
+			gettimeofday(&end,NULL);
 			break;
 		}
 
-- 
2.41.0

