From 1e55b22a0076fe66835115954b482dbc1b0d3da7 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 29 Nov 2025 22:31:16 +0100
Subject: [PATCH] remove %{arm} from excludearchs

---
 SPECS/numactl.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/numactl.spec b/SPECS/numactl.spec
index c71e544..d79cd7b 100644
--- a/SPECS/numactl.spec
+++ b/SPECS/numactl.spec
@@ -1,7 +1,7 @@
 Name:		numactl
 Summary:	Library for tuning for Non Uniform Memory Access machines
 Version:	2.0.19
-Release:	3%{dist}
+Release:	3%{dist}.redsleeve
 # libnuma is LGPLv2 and GPLv2
 # numactl binaries are GPLv2 only
 License:	LGPL-2.1-only and GPL-2.0-only
@@ -11,7 +11,7 @@ Source0:	%{url}/releases/download/v%{version}/%{name}-%{version}.tar.gz
 BuildRequires: make
 BuildRequires:	libtool automake autoconf
 
-ExcludeArch: s390 %{arm}
+ExcludeArch: s390
 
 #START INSERT
 #
@@ -112,6 +112,9 @@ rm -rf $RPM_BUILD_ROOT
 %{_mandir}/man3/*.3*
 
 %changelog
+* Sat Nov 29 2025 Jacco Ligthart <jacco@redsleeve.org>  2.0.19-3.redsleeve
+- remove %{arm} from excludearchs
+
 * Mon Nov 18 2024 Pingfan Liu <piliu@redhat.com> - 2.0.19-1
 - rebase to v2.0.19
 
-- 
2.47.3

