From c4af87bd438f86373679aed377b9e1c893094055 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Thu, 18 Aug 2022 09:13:38 +0000
Subject: [PATCH] remove %{arm} from excludearchs

---
 SPECS/numactl.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/numactl.spec b/SPECS/numactl.spec
index f42fde0..e978b48 100644
--- a/SPECS/numactl.spec
+++ b/SPECS/numactl.spec
@@ -1,7 +1,7 @@
 Name:		numactl
 Summary:	Library for tuning for Non Uniform Memory Access machines
 Version:	2.0.14
-Release:	8%{dist}
+Release:	8%{dist}.redsleeve
 # libnuma is LGPLv2 and GPLv2
 # numactl binaries are GPLv2 only
 License:	GPLv2
@@ -11,7 +11,7 @@ Source0:	%{url}/releases/download/v%{version}/%{name}-%{version}.tar.gz
 BuildRequires: make
 BuildRequires:	libtool automake autoconf
 
-ExcludeArch: s390 %{arm}
+ExcludeArch: s390
 
 #START INSERT
 #
@@ -113,6 +113,9 @@ rm -rf $RPM_BUILD_ROOT
 %{_mandir}/man3/*.3*
 
 %changelog
+* Thu Aug 18 2022 Jacco Ligthart <jacco@redsleeve.org>  2.0.14-8.redsleeve
+- remove %{arm} from excludearchs
+
 * Tue Jan  4 2022 Pingfan Liu <piliu@redhat.com> - 2.0.14-8
 - For GA release
 
-- 
2.31.1

