From cd3b740d6d1adb3afe31f66c50b28b7f2d412781 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 13:02:18 +0100
Subject: [PATCH] remove %{arm} from excludearchs

---
 SPECS/numactl.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/numactl.spec b/SPECS/numactl.spec
index 865c67c..5b39367 100644
--- a/SPECS/numactl.spec
+++ b/SPECS/numactl.spec
@@ -1,7 +1,7 @@
 Name:		numactl
 Summary:	Library for tuning for Non Uniform Memory Access machines
 Version:	2.0.16
-Release:	1%{dist}
+Release:	1%{dist}.redsleeve
 # libnuma is LGPLv2 and GPLv2
 # numactl binaries are GPLv2 only
 License:	GPLv2
@@ -11,7 +11,7 @@ Source0:	%{url}/releases/download/v%{version}/%{name}-%{version}.tar.gz
 BuildRequires: make
 BuildRequires:	libtool automake autoconf
 
-ExcludeArch: s390 %{arm}
+ExcludeArch: s390
 
 #START INSERT
 #
@@ -107,6 +107,9 @@ rm -rf $RPM_BUILD_ROOT
 %{_mandir}/man3/*.3*
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org>  2.0.16-1.redsleeve
+- remove %{arm} from excludearchs
+
 * Wed Nov 23 2022 Pingfan Liu <piliu@redhat.com> - 2.0.14-9
 - Dummy release to get s390x binary in errata
 
-- 
2.39.3

