From 3f1e16141db7a06029ff3d7eaf1a599780a5ed7c Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 26 May 2023 16:12:38 +0200
Subject: [PATCH] remove %{arm} from excludearchs

---
 SPECS/numactl.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/numactl.spec b/SPECS/numactl.spec
index b6f39ec..6a5e0db 100644
--- a/SPECS/numactl.spec
+++ b/SPECS/numactl.spec
@@ -1,7 +1,7 @@
 Name:		numactl
 Summary:	Library for tuning for Non Uniform Memory Access machines
 Version:	2.0.14
-Release:	9%{dist}
+Release:	9%{dist}.redsleeve
 # libnuma is LGPLv2 and GPLv2
 # numactl binaries are GPLv2 only
 License:	GPLv2
@@ -11,7 +11,7 @@ Source0:	%{url}/releases/download/v%{version}/%{name}-%{version}.tar.gz
 BuildRequires: make
 BuildRequires:	libtool automake autoconf
 
-ExcludeArch: s390 %{arm}
+ExcludeArch: s390
 
 #START INSERT
 #
@@ -113,6 +113,9 @@ rm -rf $RPM_BUILD_ROOT
 %{_mandir}/man3/*.3*
 
 %changelog
+* Fri May 26 2023 Jacco Ligthart <jacco@redsleeve.org>  2.0.14-9.redsleeve
+- remove %{arm} from excludearchs
+
 * Wed Nov 23 2022 Pingfan Liu <piliu@redhat.com> - 2.0.14-9
 - Dummy release to get s390x binary in errata
 
-- 
2.31.1

