From 49de371d02ed50cf937fbe24876e85587bab37a8 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 6 Jun 2026 22:14:17 +0200
Subject: [PATCH] added atomic to the linker

---
 SPECS/webkit2gtk3.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/webkit2gtk3.spec b/SPECS/webkit2gtk3.spec
index 2054011..11b6ee2 100644
--- a/SPECS/webkit2gtk3.spec
+++ b/SPECS/webkit2gtk3.spec
@@ -20,7 +20,7 @@
 
 Name:           webkit2gtk3
 Version:        2.52.3
-Release:        0%{?dist}.1
+Release:        0%{?dist}.1.redsleeve
 Summary:        GTK Web content engine library
 
 License:        LGPLv2
@@ -259,6 +259,9 @@ rm -rf Source/ThirdParty/qunit/
 %ifarch aarch64
   -DUSE_64KB_PAGE_BLOCK=ON \
 %endif
+%ifarch armv6hl
+  -DCMAKE_SHARED_LINKER_FLAGS="-latomic" \
+%endif
 %endif
   %{nil}
 
@@ -329,6 +332,9 @@ export NINJA_STATUS="[%f/%t][%e] "
 %{_datadir}/gir-1.0/JavaScriptCore-4.0.gir
 
 %changelog
+* Fri May 08 2026 Jacco Ligthart <jacco@redsleeve.org> - 2.52.3-0.1.redsleeve
+- added atomic to the linker
+
 * Mon Apr 20 2026 Michael Catanzaro <mcatanzaro@redhat.com> - 2.52.3-0.1
 - Update to 2.52.1
 
-- 
2.47.3

