Description: Only yield under ARMv7 and above (#1176)
Author: Rosen Penev <rosenp@gmail.com>
Date: Tue, 12 Nov 2019 13:56:53 -0800
Origin: upstream, https://github.com/facebook/folly/commit/62d8e6e0b91ebd6f878f3066cd9b6e5f3c18a97b.patch
Last-Update: 2021-11-24

--- ceph-16.2.6+ds.orig/src/rocksdb/third-party/folly/folly/portability/Asm.h
+++ ceph-16.2.6+ds/src/rocksdb/third-party/folly/folly/portability/Asm.h
@@ -19,7 +19,7 @@ inline void asm_volatile_pause() {
   ::_mm_pause();
 #elif defined(__i386__) || FOLLY_X64
   asm volatile("pause");
-#elif FOLLY_AARCH64 || defined(__arm__)
+#elif FOLLY_AARCH64 || (defined(__arm__) && !(__ARM_ARCH < 7))
   asm volatile("yield");
 #elif FOLLY_PPC64
   asm volatile("or 27,27,27");
