From 12aa8a77107bcd92875aac5dcbc49e9f702efbf0 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 1 Jul 2023 12:53:42 +0200
Subject: [PATCH] added arm to golang_arches

---
 SPECS/golang.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/golang.spec b/SPECS/golang.spec
index 2838719..9b6298d 100644
--- a/SPECS/golang.spec
+++ b/SPECS/golang.spec
@@ -29,7 +29,7 @@
 # Define GOROOT macros
 %global goroot          %{_prefix}/lib/%{name}
 %global gopath          %{_datadir}/gocode
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %global golibdir        %{_libdir}/%{name}
 
 # Golang build options.
@@ -102,7 +102,7 @@
 
 Name:           golang
 Version:        %{version}
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        The Go Programming Language
 # source tree includes several copies of Mark.Twain-Tom.Sawyer.txt under Public Domain
 License:        BSD and Public Domain
@@ -533,6 +533,9 @@ cd ..
 %endif
 
 %changelog
+* Sat Jul 01 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.19.10-1.redsleeve
+- added arm to golang_arches
+
 * Tue Jun 6 2023 David Benoit <dbenoit@redhat.com> - 1.19.10-1
 - Update to Go 1.19.10
 - Resolves: rhbz#2217626
-- 
2.31.1

