From 9d192c998690cee818bcf3401511398e7747c813 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 9 Aug 2022 23:16:25 +0000
Subject: [PATCH] added arm to golang_arches

---
 SPECS/golang.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/golang.spec b/SPECS/golang.spec
index 6349c68..3a3dbb2 100644
--- a/SPECS/golang.spec
+++ b/SPECS/golang.spec
@@ -29,7 +29,7 @@
 # Define GOROOT macros
 %global goroot          %{_prefix}/lib/%{name}
 %global gopath          %{_datadir}/gocode
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %global golibdir        %{_libdir}/%{name}
 
 # Golang build options.
@@ -101,7 +101,7 @@
 
 Name:           golang
 Version:        %{go_version}
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        The Go Programming Language
 # source tree includes several copies of Mark.Twain-Tom.Sawyer.txt under Public Domain
 License:        BSD and Public Domain
@@ -525,6 +525,9 @@ cd ..
 %endif
 
 %changelog
+* Wed Aug 10 2022 Jacco Ligthart <jacco@redsleeve.org> - 1.17.12-1.redsleeve
+- added arm to golang_arches
+
 * Wed Jul 20 2022 David Benoit <dbenoit@redhat.com> - 1.17.12-1
 - Update Go to version 1.17.12
 - Resolves: rhbz#2109183
-- 
2.31.1

