From 9770974fc0758092614a8aee95c51f051839030b Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 24 Nov 2023 15:29:30 +0100
Subject: [PATCH] added arm to golang_arches

---
 SPECS/golang.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/golang.spec b/SPECS/golang.spec
index b199315..4fef76b 100644
--- a/SPECS/golang.spec
+++ b/SPECS/golang.spec
@@ -29,7 +29,7 @@
 # Define GOROOT macros
 %global goroot          %{_prefix}/lib/%{name}
 %global gopath          %{_datadir}/gocode
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %global golibdir        %{_libdir}/%{name}
 
 # Golang build options.
@@ -102,7 +102,7 @@
 
 Name:           golang
 Version:        %{version}
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        The Go Programming Language
 # source tree includes several copies of Mark.Twain-Tom.Sawyer.txt under Public Domain
 License:        BSD and Public Domain
@@ -532,6 +532,9 @@ cd ..
 %endif
 
 %changelog
+* Thu Nov 02 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.19.13-1.redsleeve
+- added arm to golang_arches
+
 * Thu Oct 12 2023 Derek Parker <deparker@redhat.com> - 1.19.13-1
 - Fix CVE-2023-39325
 - Resolves: RHEL-12622
-- 
2.39.3

