From ca20ad9edf7c6d248c62687ede40679531ff0d89 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:22:21 +0100
Subject: [PATCH] added arm to golang_arches

---
 SPECS/golang.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/golang.spec b/SPECS/golang.spec
index fc58d14..5d85909 100644
--- a/SPECS/golang.spec
+++ b/SPECS/golang.spec
@@ -29,7 +29,7 @@
 # Define GOROOT macros
 %global goroot          %{_prefix}/lib/%{name}
 %global gopath          %{_datadir}/gocode
-%global golang_arches   x86_64 aarch64 ppc64le s390x
+%global golang_arches   x86_64 aarch64 ppc64le s390x %{arm}
 %global golibdir        %{_libdir}/%{name}
 
 # Golang build options.
@@ -99,7 +99,7 @@
 
 Name: golang
 Version: %{version}
-Release: 1%{?dist}.0.1
+Release: 1%{?dist}.redsleeve
 Summary: The Go Programming Language
 # source tree includes several copies of Mark.Twain-Tom.Sawyer.txt under Public Domain
 License: BSD and Public Domain
@@ -527,6 +527,9 @@ cd ..
 %files -n go-toolset
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.20.10-1.redsleeve
+- added arm to golang_arches
+
 * Fri Oct 13 2023 David Benoit <dbenoit@redhat.com> - 1.20.10-1
 - Update to Go 1.20.10
 - Fix CVE-2023-39325
-- 
2.39.3

