From 1a574570e2a9c94a598c82cd61f854b77cd3c418 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 2 Sep 2022 21:41:04 +0000
Subject: [PATCH] added -latomic for linking

---
 SPECS/librdkafka.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/SPECS/librdkafka.spec b/SPECS/librdkafka.spec
index 7949664..57ae400 100644
--- a/SPECS/librdkafka.spec
+++ b/SPECS/librdkafka.spec
@@ -1,6 +1,6 @@
 Name:		librdkafka
 Version:	1.6.1
-Release:	102%{?dist}
+Release:	102%{?dist}.redsleeve
 Summary:	The Apache Kafka C library
 
 License:	BSD
@@ -57,6 +57,9 @@ using librdkafka.
     --enable-lz4-ext \
     --enable-ssl \
     --enable-gssapi \
+%ifarch %{arm}
+    -LDFLAGS="-latomic" \
+%endif
     --enable-sasl
 
 %make_build
@@ -87,6 +90,9 @@ find %{buildroot} -name '*-static.pc' -delete -print
 
 
 %changelog
+* Fri Sep 02 2022 Jacco Ligthart <jacco@redsleeve.org> - 1.6.1-102.redsleeve
+- added -latomic for linking
+
 * Tue Feb 08 2022 Sergio Arroutbi <sarroutb@redhat.com> - 1.6.1-102
 - Changes for tests to compile and run appropriately
   Related: rhbz#2032923
-- 
2.31.1

