From b419550307c3fcab2e0c12cc2b0f0ad072b3ce25 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 27 Sep 2024 21:07:57 +0200
Subject: [PATCH] removed valgrind

---
 SPECS/systemd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 868798f..1406133 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -25,7 +25,7 @@
 Name: systemd
 Url: https://systemd.io
 Version: 252
-Release: 32%{?dist}.7
+Release: 32%{?dist}.7.redsleeve
 # For a breakdown of the licensing, see README
 License: LGPLv2+ and MIT and GPLv2+
 Summary: System and Service Manager
@@ -878,7 +878,7 @@ BuildRequires: libseccomp-devel
 BuildRequires: meson >= 0.43
 BuildRequires: gettext
 # We use RUNNING_ON_VALGRIND in tests, so the headers need to be available
-BuildRequires: valgrind-devel
+#BuildRequires: valgrind-devel
 BuildRequires: pkgconfig(bash-completion)
 BuildRequires: pkgconfig(tss2-esys)
 BuildRequires: pkgconfig(tss2-rc)
@@ -1700,6 +1700,9 @@ systemd-hwdb update &>/dev/null || :
 %{_prefix}/lib/dracut/modules.d/70rhel-net-naming-sysattrs/*
 
 %changelog
+* Fri Sep 27 2024 Jacco Ligthart <jacco@redsleeve.org> - 252-32.7.redsleeve
+- removed valgrind
+
 * Tue Sep 03 2024 Release Engineering <releng@rockylinux.org> - 252-32
 - Set support URL to the wiki
 - Set sbat mail to security@rockylinux.org
-- 
2.43.5

