From f964f0c05872e13e07e38840f36f6f329a8ab036 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 13:14:30 +0100
Subject: [PATCH] removed valgrind

---
 SPECS/systemd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 778682a..6ed01f8 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -21,7 +21,7 @@
 Name:           systemd
 Url:            https://systemd.io
 Version:        252
-Release:        18%{?dist}
+Release:        18%{?dist}.redsleeve
 # For a breakdown of the licensing, see README
 License:        LGPLv2+ and MIT and GPLv2+
 Summary:        System and Service Manager
@@ -502,7 +502,7 @@ BuildRequires:  libseccomp-devel
 BuildRequires:  meson >= 0.43
 BuildRequires:  gettext
 # We use RUNNING_ON_VALGRIND in tests, so the headers need to be available
-BuildRequires:  valgrind-devel
+#BuildRequires:  valgrind-devel
 BuildRequires:  pkgconfig(bash-completion)
 BuildRequires:  pkgconfig(tss2-esys)
 BuildRequires:  pkgconfig(tss2-rc)
@@ -1268,6 +1268,9 @@ getent passwd systemd-oom &>/dev/null || useradd -r -l -g systemd-oom -d / -s /s
 %files standalone-sysusers -f .file-list-standalone-sysusers
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 252-18.redsleeve
+- removed valgrind
+
 * Tue Aug 22 2023 systemd maintenance team <systemd-maint@redhat.com> - 252-18
 - doc: add downstream CONTRIBUTING document (#2170883)
 - doc: improve CONTRIBUTING document (#2170883)
-- 
2.39.3

