From 77f30270ea50fc766fe5e77f7685e2e57ab75fa0 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 4 Mar 2023 20:15:12 +0000
Subject: [PATCH] removed valgrind

---
 SPECS/systemd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 717927e..1779dac 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -21,7 +21,7 @@
 Name:           systemd
 Url:            https://www.freedesktop.org/wiki/Software/systemd
 Version:        250
-Release:        12%{?dist}.3
+Release:        12%{?dist}.3.redsleeve
 # For a breakdown of the licensing, see README
 License:        LGPLv2+ and MIT and GPLv2+
 Summary:        System and Service Manager
@@ -492,7 +492,7 @@ BuildRequires:  libseccomp-devel
 BuildRequires:  meson >= 0.43
 BuildRequires:  gettext
 # We use RUNNING_ON_VALGRIND in tests, so the headers need to be available
-BuildRequires:  valgrind-devel
+#BuildRequires:  valgrind-devel
 BuildRequires:  pkgconfig(bash-completion)
 BuildRequires:  pkgconfig(tss2-esys)
 BuildRequires:  pkgconfig(tss2-rc)
@@ -1222,6 +1222,9 @@ getent passwd systemd-oom &>/dev/null || useradd -r -l -g systemd-oom -d / -s /s
 %files standalone-sysusers -f .file-list-standalone-sysusers
 
 %changelog
+* Sat Mar 04 2023 Jacco Ligthart <jacco@redsleeve.org> - 250-12.1.redsleeve
+- removed valgrind
+
 * Mon Feb 13 2023 systemd maintenance team <systemd-maint@redhat.com> - 250-12.3
 - shared/json: allow json_variant_dump() to return an error (#2149074)
 - shared/json: use different return code for empty input (#2149074)
-- 
2.31.1

