From ee800630eadd6faf0b4965e4268d01dfbe377b69 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 27 Sep 2022 20:14:12 +0000
Subject: [PATCH] removed valgrind

---
 SPECS/systemd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 631742e..aeeeb37 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -21,7 +21,7 @@
 Name:           systemd
 Url:            https://www.freedesktop.org/wiki/Software/systemd
 Version:        250
-Release:        6%{?dist}.1
+Release:        6%{?dist}.1.redsleeve
 # For a breakdown of the licensing, see README
 License:        LGPLv2+ and MIT and GPLv2+
 Summary:        System and Service Manager
@@ -188,7 +188,7 @@ BuildRequires:  libseccomp-devel
 BuildRequires:  meson >= 0.43
 BuildRequires:  gettext
 # We use RUNNING_ON_VALGRIND in tests, so the headers need to be available
-BuildRequires:  valgrind-devel
+#BuildRequires:  valgrind-devel
 BuildRequires:  pkgconfig(bash-completion)
 BuildRequires:  pkgconfig(tss2-esys)
 BuildRequires:  pkgconfig(tss2-rc)
@@ -918,6 +918,9 @@ getent passwd systemd-oom &>/dev/null || useradd -r -l -g systemd-oom -d / -s /s
 %files standalone-sysusers -f .file-list-standalone-sysusers
 
 %changelog
+* Tue Sep 27 2022 Jacco Ligthart <jacco@redsleeve.org> - 250-6.1.redsleeve
+- removed valgrind
+
 * Wed Jul 20 2022 systemd maintenance team <systemd-maint@redhat.com> - 250-6.1
 - shutdown: get only active md arrays. (#2087677)
 - test: lvm 2.03.15 dropped the static autoactivation (#2087677)
-- 
2.31.1

