--- a/clang/lib/Driver/ToolChains/Gnu.cpp-bak	2023-06-04 16:48:04.389190068 +0200
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp	2023-06-04 16:50:14.327438386 +0200
@@ -1974,6 +1974,7 @@
 static const char *ConvertToGccTriple(StringRef CandidateTriple) {
   return llvm::StringSwitch<const char *>(CandidateTriple)
       .Case("aarch64-redhat-linux-gnu", "aarch64-redhat-linux")
+      .Case("armv6l-redhat-linux-gnueabihf", "armv6hl-redhat-linux-gnueabi")
       .Case("i686-redhat-linux-gnu", "i686-redhat-linux")
       .Case("ppc64le-redhat-linux-gnu", "ppc64le-redhat-linux")
       .Case("s390x-redhat-linux-gnu", "s390x-redhat-linux")
