From 252ae0fe6743f35b965f2968c7e010d0aa94da07 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 9 Jun 2025 13:47:10 +0200
Subject: [PATCH] fixed builddeps for arm

---
 SPECS/libnbd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libnbd.spec b/SPECS/libnbd.spec
index 153bf98..662e8a2 100644
--- a/SPECS/libnbd.spec
+++ b/SPECS/libnbd.spec
@@ -21,7 +21,7 @@
 
 Name:           libnbd
 Version:        1.20.3
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        NBD client library in userspace
 
 License:        LGPL-2.0-or-later AND BSD-3-Clause
@@ -101,7 +101,7 @@ BuildRequires:  util-linux
 # nbdkit for i686.  These are only needed for the test suite so make
 # them optional.  This reduces our test exposure on 32 bit platforms,
 # although there is still Fedora/armv7 and some upstream testing.
-%ifnarch %{ix86}
+%ifnarch %{ix86} %{arm}
 BuildRequires:  qemu-img
 BuildRequires:  nbdkit
 BuildRequires:  nbdkit-data-plugin
@@ -390,6 +390,9 @@ make %{?_smp_mflags} check || {
 
 
 %changelog
+* Mon Jun 09 2025 Jacco Ligthart <jacco@redsleeve.org> - 1.20.3-1.redsleeve
+- fixed builddeps for arm
+
 * Sat Sep 28 2024 Richard W.M. Jones <rjones@redhat.com> - 1.20.3-1
 - Rebase to libnbd 1.20.3
 
-- 
2.43.5

