From 694ca184db6dedd81de878ee92ff75d68066d2eb Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:25:51 +0100
Subject: [PATCH] fixed builddeps for arm

---
 SPECS/libnbd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libnbd.spec b/SPECS/libnbd.spec
index e9bfc4c..a467978 100644
--- a/SPECS/libnbd.spec
+++ b/SPECS/libnbd.spec
@@ -12,7 +12,7 @@
 
 Name:           libnbd
 Version:        1.16.0
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        NBD client library in userspace
 
 License:        LGPLv2+
@@ -86,7 +86,7 @@ BuildRequires:  util-linux
 # nbdkit for i686.  These are only needed for the test suite so make
 # them optional.  This reduces our test exposure on 32 bit platforms,
 # although there is still Fedora/armv7 and some upstream testing.
-%ifnarch %{ix86}
+%ifnarch %{ix86} %{arm}
 BuildRequires:  qemu-img
 BuildRequires:  nbdkit
 BuildRequires:  nbdkit-data-plugin
@@ -364,6 +364,9 @@ make %{?_smp_mflags} check || {
 
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 1.16.0-1.redsleeve
+- fixed builddeps for arm
+
 * Tue Apr 18 2023 Richard W.M. Jones <rjones@redhat.com> - 1.16.0-1
 - Rebase to 1.16.0
   resolves: rhbz#2168628
-- 
2.39.3

