From 700f62c71faa56fae1899f16c356ad3505f6f5db Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Tue, 27 Sep 2022 21:06:42 +0000
Subject: [PATCH] fixed builddeps for arm

---
 SPECS/libnbd.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/libnbd.spec b/SPECS/libnbd.spec
index b1800b5..bc3b1be 100644
--- a/SPECS/libnbd.spec
+++ b/SPECS/libnbd.spec
@@ -9,7 +9,7 @@
 
 Name:           libnbd
 Version:        1.10.5
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        NBD client library in userspace
 
 License:        LGPLv2+
@@ -78,7 +78,7 @@ BuildRequires:  util-linux
 # nbdkit for i686.  These are only needed for the test suite so make
 # them optional.  This reduces our test exposure on 32 bit platforms,
 # although there is still Fedora/armv7 and some upstream testing.
-%ifnarch %{ix86}
+%ifnarch %{ix86} %{arm}
 BuildRequires:  qemu-img
 BuildRequires:  nbdkit
 BuildRequires:  nbdkit-data-plugin
@@ -323,6 +323,9 @@ make %{?_smp_mflags} check || {
 
 
 %changelog
+* Tue Sep 27 2022 Jacco Ligthart <jacco@redsleeve.org> - 1.10.5-1.redsleeve
+- fixed builddeps for arm
+
 * Thu Feb 10 2022 Richard W.M. Jones <rjones@redhat.com> - 1.10.5-1
 - Rebase to new stable branch version 1.10.5
   resolves: rhbz#2011708
-- 
2.31.1

