From 1bb9cf68b51997c62cdb79545999f57c218f919f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Fri, 31 May 2024 17:27:32 +0200
Subject: [PATCH] changed armv7 to arm

---
 SPECS/zziplib.spec | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/SPECS/zziplib.spec b/SPECS/zziplib.spec
index ba18d9c..dadd7d9 100644
--- a/SPECS/zziplib.spec
+++ b/SPECS/zziplib.spec
@@ -1,7 +1,7 @@
 Summary: Lightweight library to easily extract data from zip files
 Name: zziplib
 Version: 0.13.71
-Release: 11%{?dist}
+Release: 11%{?dist}.redsleeve
 License: LGPLv2+ or MPLv1.1
 URL: http://zziplib.sourceforge.net/
 #Source: https://github.com/gdraheim/zziplib/archive/v%{version}.tar.gz
@@ -91,10 +91,10 @@ sed -i 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' libtool
 # These platforms have a correct _config.h already
 
 pushd %{_builddir}/zziplib-%{version}
-%ifarch i686 armv7hl
+%ifarch i686 %{arm}
  patch -p2 < %{PATCH100}
 %endif
-%ifnarch i686 armv7hl
+%ifnarch i686 %{arm}
  patch -p2 < %{PATCH101}
 %endif
 popd
@@ -124,6 +124,9 @@ popd
 %{_mandir}/man3/*
 
 %changelog
+* Fri May 31 2024 Jacco Ligthart <jacco@redsleeve.org> - 0.13.71-11.redsleeve
+- changed armv7 to arm
+
 * Wed Feb 28 2024 Jakub Martisko <jamartis@redhat.com> - 0.13.71-11
 - Fix CVE-2020-18770
   Previous patch was causing segfault
-- 
2.39.3

