From 6bb5e0fc2826b97ea5f835149164e291d63ea0e1 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Mon, 5 Dec 2022 21:25:52 +0000
Subject: [PATCH] added armv6 to rust_arches

---
 SPECS/rust.spec | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/SPECS/rust.spec b/SPECS/rust.spec
index 0c8bbe0..971d3f2 100644
--- a/SPECS/rust.spec
+++ b/SPECS/rust.spec
@@ -1,6 +1,6 @@
 # Only x86_64 and i686 are Tier 1 platforms at this time.
 # https://doc.rust-lang.org/nightly/rustc/platform-support.html
-%global rust_arches x86_64 i686 aarch64 ppc64le s390x
+%global rust_arches x86_64 i686 aarch64 ppc64le s390x armv6hl
 
 # The channel can be stable, beta, or nightly
 %{!?channel: %global channel stable}
@@ -84,7 +84,7 @@
 
 Name:           rust
 Version:        1.62.1
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 Summary:        The Rust Programming Language
 License:        (ASL 2.0 or MIT) and (BSD and MIT)
 # ^ written as: (rust itself) and (bundled libraries)
@@ -129,6 +129,9 @@ Patch102:       rustc-1.58.0-no-default-pie.patch
   if arch == "armv7hl" then
     arch = "armv7"
     abi = "gnueabihf"
+  elseif arch == "armv6hl" then
+    arch = "arm"
+    abi = "gnueabihf"
   elseif arch == "ppc64" then
     arch = "powerpc64"
   elseif arch == "ppc64le" then
@@ -1032,6 +1035,9 @@ end}
 
 
 %changelog
+* Mon Dec 05 2022 Jacco Ligthart <jacco@redsleeve.org> - 1.62.1-1.redsleeve
+- added armv6 to rust_arches
+
 * Tue Jul 19 2022 Josh Stone <jistone@redhat.com> - 1.62.1-1
 - Update to 1.62.1.
 
-- 
2.31.1

