From 561459e57a41b3372e6fff00b86c9b4d1dd70967 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 19 Jan 2025 13:28:32 +0100
Subject: [PATCH] added armv6 to rust_arches

---
 SPECS/rust.spec | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/SPECS/rust.spec b/SPECS/rust.spec
index 6bd1a47..8b36f05 100644
--- a/SPECS/rust.spec
+++ b/SPECS/rust.spec
@@ -1,6 +1,6 @@
 Name:           rust
 Version:        1.79.0
-Release:        2%{?dist}
+Release:        2%{?dist}.redsleeve
 Summary:        The Rust Programming Language
 License:        (Apache-2.0 OR MIT) AND (Artistic-2.0 AND BSD-3-Clause AND ISC AND MIT AND MPL-2.0 AND Unicode-DFS-2016)
 # ^ written as: (rust itself) and (bundled libraries)
@@ -8,7 +8,7 @@ URL:            https://www.rust-lang.org
 
 # Only x86_64, i686, and aarch64 are Tier 1 platforms at this time.
 # https://doc.rust-lang.org/nightly/rustc/platform-support.html
-%global rust_arches x86_64 i686 aarch64 ppc64le s390x
+%global rust_arches x86_64 i686 aarch64 ppc64le s390x armv6hl
 ExclusiveArch:  %{rust_arches}
 
 # To bootstrap from scratch, set the channel and date from src/stage0.json
@@ -179,6 +179,9 @@ Patch100:       rustc-1.79.0-disable-libssh2.patch
   if arch == "armv7hl" then
     arch = "armv7"
     abi = "gnueabihf"
+  elseif arch == "armv6hl" then
+    arch = "arm"
+    abi = "gnueabihf"
   elseif arch == "ppc64" then
     arch = "powerpc64"
   elseif arch == "ppc64le" then
@@ -787,7 +790,7 @@ fi
 %define profiler /usr/lib/clang/%{llvm_compat_version}/lib/%{_arch}-redhat-linux-gnu/libclang_rt.profile.a
 %else
 %if 0%{?clang_major_version} >= 17
-%define profiler %{clang_resource_dir}/lib/%{_arch}-redhat-linux-gnu/libclang_rt.profile.a
+%define profiler %{clang_resource_dir}/lib/%{_arch}-redhat-linux-gnueabihf/libclang_rt.profile.a
 %else
 # The exact profiler path is version dependent..
 %define profiler %(echo %{_libdir}/clang/??/lib/libclang_rt.profile-*.a)
@@ -1146,6 +1149,9 @@ rm -rf "./build/%{rust_triple}/stage2-tools/%{rust_triple}/cit/"
 
 
 %changelog
+* Mon Dec 23 2024 Jacco Ligthart <jacco@redsleeve.org> - 1.79.0-2.redsleeve
+- added armv6 to rust_arches
+
 * Tue Aug 13 2024 Josh Stone <jistone@redhat.com> - 1.79.0-2
 - Disable jump threading of float equality
 
-- 
2.43.5

