--- a/gcc/testsuite/c-c++-common/goacc/kernels-decompose-pr100400-1-3.c
+++ b/gcc/testsuite/c-c++-common/goacc/kernels-decompose-pr100400-1-3.c
@@ -1,6 +1,6 @@
 /* { dg-additional-options "--param openacc-kernels=decompose" } */

-/* { dg-additional-options "-fchecking" }
+/* { dg-additional-options "-fchecking -fno-report-bug" }
    { dg-ice TODO { c++ } }
    { dg-prune-output "during GIMPLE pass: omp_oacc_kernels_decompose" } */

--- a/gcc/testsuite/c-c++-common/goacc/kernels-decompose-pr100400-1-4.c
+++ b/gcc/testsuite/c-c++-common/goacc/kernels-decompose-pr100400-1-4.c
@@ -1,6 +1,6 @@
 /* { dg-additional-options "--param openacc-kernels=decompose" } */

-/* { dg-additional-options "-fchecking" }
+/* { dg-additional-options "-fchecking -fno-report-bug" }
    { dg-ice TODO { c++ } }
    { dg-prune-output "during GIMPLE pass: omp_oacc_kernels_decompose" } */

diff --git a/gcc/testsuite/g++.dg/cpp0x/vt-88982.C b/gcc/testsuite/g++.dg/cpp0x/vt-88982.C
index 7a868233d73..45307a63483 100644
--- a/gcc/testsuite/g++.dg/cpp0x/vt-88982.C
+++ b/gcc/testsuite/g++.dg/cpp0x/vt-88982.C
@@ -1,6 +1,6 @@
 // PR c++/88982
 // { dg-do compile { target c++11 } }
-// { dg-additional-options "-fchecking" }
+// { dg-additional-options "-fchecking -fno-report-bug" }
 // { dg-ice "tsubst_pack_expansion" }
 
 template<typename...Ts> struct A {
diff --git a/gcc/testsuite/g++.dg/cpp1y/auto-fn61.C b/gcc/testsuite/g++.dg/cpp1y/auto-fn61.C
index bed5ea0cfc0..cb235bf891a 100644
--- a/gcc/testsuite/g++.dg/cpp1y/auto-fn61.C
+++ b/gcc/testsuite/g++.dg/cpp1y/auto-fn61.C
@@ -1,6 +1,6 @@
 // PR c++/88003
 // { dg-do compile { target c++14 } }
-// { dg-additional-options "-fchecking" }
+// { dg-additional-options "-fchecking -fno-report-bug" }
 // { dg-ice "poplevel_class" }
 
 auto test() {
diff --git a/gcc/testsuite/g++.dg/cpp1z/constexpr-lambda26.C b/gcc/testsuite/g++.dg/cpp1z/constexpr-lambda26.C
index 0cdb400d21c..8f17cca31dd 100644
--- a/gcc/testsuite/g++.dg/cpp1z/constexpr-lambda26.C
+++ b/gcc/testsuite/g++.dg/cpp1z/constexpr-lambda26.C
@@ -1,6 +1,6 @@
 // PR c++/87765
 // { dg-do compile { target c++17 } }
-// { dg-additional-options "-fchecking" }
+// { dg-additional-options "-fchecking -fno-report-bug" }
 // { dg-ice "cxx_eval_constant_expression" }
 
 template <int N>
