From 441ad77a0c89f4abdf104c668d579607f338a052 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 29 May 2012 20:44:18 +0100
Subject: [PATCH 2/6] configure: Allow user defined C compiler flags.

---
 configure.ac | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index aa5f26f28..5153853b7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -609,6 +609,10 @@ AS_CASE([$host],
       internal_cflags="$gcc_warnings"],
     [common_cflags="-O"])])
 
+# Allow CFLAGS and LDFLAGS to be added.
+common_cflags="$common_cflags $CFLAGS"
+cclibs="$cclibs $LDFLAGS"
+
 internal_cppflags="-DCAML_NAME_SPACE $internal_cppflags"
 
 # Enable SSE2 on x86 mingw to avoid using 80-bit registers.
-- 
2.32.0

