--- ./a/zzip/_config.h	2021-07-21 14:18:09.000000000 +0200
+++ ./b/zzip/_config.h	2021-07-21 14:46:24.037432969 +0200
@@ -138,9 +138,12 @@
 
 /* whether the system defaults to 32bit off_t but can do 64bit when requested
    */
+#if __WORDSIZE == 32
 #ifndef ZZIP_LARGEFILE_SENSITIVE 
 #define ZZIP_LARGEFILE_SENSITIVE  1 
 #endif
+#endif
+/* #undef LARGEFILE_SENSITIVE */
 
 /* Define to the sub-directory where libtool stores uninstalled libraries. */
 #ifndef ZZIP_LT_OBJDIR 
@@ -227,9 +230,13 @@
 #endif
 
 /* Number of bits in a file offset, on hosts where this is settable. */
+#if __WORDSIZE == 32
 #ifndef ZZIP__FILE_OFFSET_BITS 
 #define ZZIP__FILE_OFFSET_BITS  64 
 #endif
+#endif
+/* #undef _FILE_OFFSET_BITS */
+
 
 /* Define for large files, on AIX-style hosts. */
 /* #undef _LARGE_FILES */
