From e220185b537608f340d0a92c303fc9346842d6c2 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 21 Aug 2022 20:37:14 +0000
Subject: [PATCH] no valgrind and memchecker for %{arm}

---
 SPECS/openmpi.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/openmpi.spec b/SPECS/openmpi.spec
index a763dcd..8f8bfaf 100644
--- a/SPECS/openmpi.spec
+++ b/SPECS/openmpi.spec
@@ -31,7 +31,7 @@
 Name:           openmpi%{?_cc_name_suffix}
 Epoch:          1
 Version:        4.1.1
-Release:        5%{?dist}
+Release:        5%{?dist}.redsleeve
 Summary:        Open Message Passing Interface
 License:        BSD and MIT and Romio
 URL:            http://www.open-mpi.org/
@@ -53,7 +53,9 @@ BuildRequires:  libtool
 BuildRequires:  perl(Data::Dumper)
 BuildRequires:  perl(File::Find)
 %endif
+%ifnarch %{arm}
 BuildRequires:  valgrind-devel
+%endif
 %if %{with rdma}
 BuildRequires:  opensm-devel > 3.3.0
 BuildRequires:  rdma-core-devel
@@ -187,8 +189,10 @@ OpenMPI support for Python 3.
 	--enable-mpi-java \
 	--enable-mpi1-compatibility \
 	--with-sge \
+%ifnarch %{arm}
 	--with-valgrind \
 	--enable-memchecker \
+%endif
 	--with-hwloc=/usr \
 %if !0%{?el7}
 	--with-libevent=external \
@@ -358,6 +362,9 @@ make check
 
 
 %changelog
+* Sun Aug 21 2022 Jacco Ligthart <jacco@redsleeve.org> - 4.1.1-5.redsleeve
+- no valgrind and memchecker for %{arm}
+
 * Thu Feb 17 2022 Honggang Li <honli@redhat.com> - 4.1.1-5
 - Revert v4.1.2
 - Add Epoch tag
-- 
2.31.1

