From 7cdbb9b36f200d11e4acc62e2efd964f8abe3d25 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:44:06 +0100
Subject: [PATCH] no valgrind and memchecker for %{arm}

---
 SPECS/openmpi.spec | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/SPECS/openmpi.spec b/SPECS/openmpi.spec
index 17dbcdf..ebeee42 100644
--- a/SPECS/openmpi.spec
+++ b/SPECS/openmpi.spec
@@ -31,7 +31,7 @@
 Name:           openmpi%{?_cc_name_suffix}
 Epoch:          2
 Version:        4.1.1
-Release:        7%{?dist}
+Release:        7%{?dist}.redsleeve
 Summary:        Open Message Passing Interface
 License:        BSD and MIT and Romio
 URL:            http://www.open-mpi.org/
@@ -53,7 +53,9 @@ BuildRequires:  libtool
 BuildRequires:  perl(Data::Dumper)
 BuildRequires:  perl(File::Find)
 %endif
+%ifnarch %{arm}
 BuildRequires:  valgrind-devel
+%endif
 %if %{with rdma}
 BuildRequires:  opensm-devel > 3.3.0
 BuildRequires:  rdma-core-devel
@@ -187,8 +189,10 @@ OpenMPI support for Python 3.
 	--enable-mpi-java \
 	--enable-mpi1-compatibility \
 	--with-sge \
+%ifnarch %{arm}
 	--with-valgrind \
 	--enable-memchecker \
+%endif
 	--with-hwloc=/usr \
 %if !0%{?el7}
 	--with-libevent=external \
@@ -358,6 +362,9 @@ make check
 
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 2:4.1.1-7.redsleeve
+- no valgrind and memchecker for %{arm}
+
 * Fri Jul 21 2023 Kamal Heib <kheib@redhat.com> - 2:4.1.1-7
 - Increase Epoch tag
 - Resolves: rhbz#2159637, rhbz#2218121, rhbz#2216042
-- 
2.39.3

