From a0381b36fe49fda56d14673d2e5bfb1a96ca7e5f Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 29 Nov 2025 22:32:54 +0100
Subject: [PATCH] apply definitions_for_user_vfp also for armv6

---
 SPECS/polkit.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/polkit.spec b/SPECS/polkit.spec
index fd6c487..abc8018 100644
--- a/SPECS/polkit.spec
+++ b/SPECS/polkit.spec
@@ -22,7 +22,7 @@
 Summary: An authorization framework
 Name: polkit
 Version: 0.117
-Release: 14%{?dist}
+Release: 14%{?dist}.redsleeve
 License: LGPLv2+
 URL: http://www.freedesktop.org/wiki/Software/polkit
 Source0: http://www.freedesktop.org/software/polkit/releases/%{name}-%{version}.tar.gz
@@ -205,7 +205,7 @@ pushd firefox-%{mozjs_version}
 %patch14 -p1
 %patch15 -p1
 
-%ifarch armv7hl
+%ifarch %{arm}
 # Include definitions for user vfp on armv7 as it causes the compilation to fail without them
 # https://bugzilla.mozilla.org/show_bug.cgi?id=1526653
 %patch17 -p1
@@ -400,6 +400,9 @@ exit 0
 %endif
 
 %changelog
+* Sat Nov 29 2025 Jacco Ligthart <jacco@redsleeve.org> - 0.117-14.redsleeve
+- apply definitions_for_user_vfp also for armv6
+
 * Tue Aug 05 2025 Jan Rybar <jrybar@redhat.com> - 0.117-14
 - aarch64: test failure with undefined symbol: __aarch64_ldadd4_acq_rel
 - Resolves: RHEL-98662
-- 
2.47.3

