diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_date/char/12791.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_date/char/12791.cc
index fa740b42b88..5d808ee51ee 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_date/char/12791.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_date/char/12791.cc
@@ -39,14 +39,14 @@ void test01()
   const ios_base::iostate good = ios_base::goodbit;
   ios_base::iostate errorstate = good;
 
-  iss.str("62/04/71");
+  iss.str("60/04/71");
   iterator_type is_it01(iss);
   tm time01;
   errorstate = good;
   iterator_type ret01 = tim_get.get_date(is_it01, end, iss, errorstate,
 					 &time01);
   VERIFY( errorstate == ios_base::failbit );
-  VERIFY( *ret01 == '2' );
+  VERIFY( *ret01 == '6' );
 
   iss.str("04/38/71");
   iterator_type is_it02(iss);
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_date/wchar_t/12791.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_date/wchar_t/12791.cc
index 07f4c369e58..c975c8523d8 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_date/wchar_t/12791.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_date/wchar_t/12791.cc
@@ -40,14 +40,14 @@ void test01()
   const ios_base::iostate good = ios_base::goodbit;
   ios_base::iostate errorstate = good;
 
-  iss.str(L"62/04/71");
+  iss.str(L"60/04/71");
   iterator_type is_it01(iss);
   tm time01;
   errorstate = good;
   iterator_type ret01 = tim_get.get_date(is_it01, end, iss, errorstate,
 					 &time01);
   VERIFY( errorstate == ios_base::failbit );
-  VERIFY( *ret01 == L'2' );
+  VERIFY( *ret01 == L'6' );
 
   iss.str(L"04/38/71");
   iterator_type is_it02(iss);
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/1.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/1.cc
index e6f53de4a55..c9d9896a6c0 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/1.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_year/char/1.cc
@@ -76,8 +76,8 @@ void test01()
   errorstate = good;
   iterator_type ret03 = tim_get.get_year(is_it03, end, iss, errorstate,
 					 &time03);
-  VERIFY( time03.tm_year == 197 - 1900 );
-  VERIFY( errorstate == good );
+  VERIFY( time03.tm_year == 3 );
+  VERIFY( errorstate == ios_base::failbit );
   VERIFY( *ret03 == 'd' );
 
   iss.str("71d71");
diff --git a/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/1.cc b/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/1.cc
index ba95fb91520..e17712098f4 100644
--- a/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/1.cc
+++ b/libstdc++-v3/testsuite/22_locale/time_get/get_year/wchar_t/1.cc
@@ -76,8 +76,8 @@ void test01()
   errorstate = good;
   iterator_type ret03 = tim_get.get_year(is_it03, end, iss, errorstate,
 					 &time03);
-  VERIFY( time03.tm_year == 197 - 1900 );
-  VERIFY( errorstate == good );
+  VERIFY( time03.tm_year == 3 );
+  VERIFY( errorstate == ios_base::failbit );
   VERIFY( *ret03 == L'd' );
 
   iss.str(L"71d71");
