From d6c0bb069c2085ff98921abdfc108231c910bc3d Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sat, 25 Nov 2023 21:40:45 +0100
Subject: [PATCH] no valgrind for %{arm}

---
 SPECS/mpich.spec | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/SPECS/mpich.spec b/SPECS/mpich.spec
index b5d5eef..a65cc5d 100644
--- a/SPECS/mpich.spec
+++ b/SPECS/mpich.spec
@@ -1,7 +1,7 @@
 Summary:        A high-performance implementation of MPI
 Name:           mpich
 Version:        4.1.1
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 License:        MIT
 URL:            https://www.mpich.org/
 
@@ -22,7 +22,7 @@ BuildRequires:  automake
 BuildRequires:  libtool
 # For ./maint/extractcvars
 BuildRequires:  perl(lib)
-%ifnarch s390 %{mips}
+%ifnarch s390 %{mips} %{arm}
 BuildRequires:  valgrind-devel
 %endif
 # For %%{python3_sitearch}
@@ -105,9 +105,9 @@ mpich support for Python 3.
 
 %patch0 -p1
 
-%ifarch %{arm}
-%patch1 -p1
-%endif
+#ifarch %{arm}
+#patch1 -p1
+#endif
 
 %build
 ./autogen.sh
@@ -241,6 +241,9 @@ make check VERBOSE=1 \
 %{python3_sitearch}/%{name}.pth
 
 %changelog
+* Sat Nov 25 2023 Jacco Ligthart <jacco@redsleeve.org> - 4.1.1-1.redsleeve
+- no valgrind for %{arm}
+
 * Sat Jun 03 2023 Kamal Heib <kheib@redhat.com> - 4.1.1-1
 - Update to upstream release 4.1.1
 - Resolves: rhbz#2212010
-- 
2.39.3

