From b1d8ae81f2f5ab4d2dc1c5c0066212a98435ec92 Mon Sep 17 00:00:00 2001
From: Jacco Ligthart <jacco@redsleeve.org>
Date: Sun, 21 Aug 2022 20:28:34 +0000
Subject: [PATCH] no valgrind for %{arm}

---
 SPECS/mpich.spec | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/SPECS/mpich.spec b/SPECS/mpich.spec
index 4ad1ea3..8ced680 100644
--- a/SPECS/mpich.spec
+++ b/SPECS/mpich.spec
@@ -1,7 +1,7 @@
 Summary:        A high-performance implementation of MPI
 Name:           mpich
 Version:        3.4.2
-Release:        1%{?dist}
+Release:        1%{?dist}.redsleeve
 License:        MIT
 URL:            https://www.mpich.org/
 
@@ -28,7 +28,7 @@ BuildRequires:  automake
 BuildRequires:  libtool
 # For ./maint/extractcvars
 BuildRequires:  perl(lib)
-%ifnarch s390 %{mips}
+%ifnarch s390 %{mips} %{arm}
 BuildRequires:  valgrind-devel
 %endif
 # For %%{python3_sitearch}
@@ -251,6 +251,9 @@ make check VERBOSE=1 \
 %{python3_sitearch}/%{name}.pth
 
 %changelog
+* Sun Aug 21 2022 Jacco Ligthart <jacco@redsleeve.org> - 3.4.2-1.redsleeve
+- no valgrind for %{arm}
+
 * Sat Nov 27 2021 Honggang Li <honli@redhat.com> - 3.4.2-1
 - Update to latest version 3.4.2
 - Related: rhbz#2015398
-- 
2.31.1

