From 4f2786d7457efbc95171f2742bbcc49b5baf3933 Mon Sep 17 00:00:00 2001
From: Mustafa Gezen <mustafa@rockylinux.org>
Date: Thu, 2 Jun 2022 17:58:10 +0200
Subject: [PATCH] Add tls_version1 as unstable test

---
 mysql-test/unstable-tests | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mysql-test/unstable-tests b/mysql-test/unstable-tests
index c22ded07..c8b02e1d 100644
--- a/mysql-test/unstable-tests
+++ b/mysql-test/unstable-tests
@@ -100,6 +100,7 @@ main.information_schema                          : Modified in 10.5.9
 main.information_schema_all_engines              : MDEV-23421 - Wrong result
 main.innodb_ext_key                              : MDEV-20169 - Wrong result
 main.innodb_icp                                  : MDEV-20168 - Wrong execution plans
+main.innodb_ext_key                              : MDEV-22119 - fails with NULL values
 main.innodb_mrr_cpk                              : MDEV-24737 - Server crash
 main.invisible_field_grant_completely            : MDEV-22254 - Syscall param write points to uninitialised bytes
 main.ipv4_and_ipv6                               : MDEV-20964 - Wrong result
@@ -207,8 +208,10 @@ main.subselect_innodb                            : MDEV-10614 - Wrong result
 main.symlink-myisam-11902                        : Modified in 10.5.9
 main.table_value_constr                          : Modified in 10.5.9
 main.tc_heuristic_recover                        : MDEV-14189 - Wrong result
+main.thread_pool_info                            : MDEV-20372 - thread_pool_info fails randomly in 10.5
 main.temp_table_symlink                          : MDEV-24058 - Wrong error code
 main.tls_version                                 : MDEV-23417 - Wrong result
+main.tls_version1                                : MDEV-21965 - old TLS versions don't work on latest Debian and Ubuntu releases
 main.type_blob                                   : MDEV-15195 - Wrong result
 main.type_datetime_hires                         : MDEV-10687 - Timeout
 main.type_float                                  : MDEV-20532 - Floating point differences
-- 
2.32.0 (Apple Git-132)

